==167396== Memcheck, a memory error detector
==167396== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==167396== Using Valgrind-3.18.1 and LibVEX; rerun with -h for copyright info
==167396== Command: python3 test_fft.py
==167396== Parent PID: 584
==167396== 
==167396== Conditional jump or move depends on uninitialised value(s)
==167396==    at 0x36B0F7: ??? (in /usr/bin/python3.10)
==167396==    by 0x29EF2B: ??? (in /usr/bin/python3.10)
==167396==    by 0x3605FC: ??? (in /usr/bin/python3.10)
==167396==    by 0x29E779: ??? (in /usr/bin/python3.10)
==167396==    by 0x29C84F: ??? (in /usr/bin/python3.10)
==167396==    by 0x2B2973: ??? (in /usr/bin/python3.10)
==167396==    by 0x2B257A: ??? (in /usr/bin/python3.10)
==167396==    by 0x2B20D3: ??? (in /usr/bin/python3.10)
==167396==    by 0x2B0938: ??? (in /usr/bin/python3.10)
==167396==    by 0x2B014C: ??? (in /usr/bin/python3.10)
==167396==    by 0x2AE53C: ??? (in /usr/bin/python3.10)
==167396==    by 0x2ADD22: ??? (in /usr/bin/python3.10)
==167396== 
==167396== Warning: set address range perms: large range [0x640af000, 0x75eac000) (noaccess)
==167396== Warning: set address range perms: large range [0x64200000, 0x75dfd000) (defined)
==167396== Warning: set address range perms: large range [0x64db0000, 0x75b52000) (defined)
==167396== Invalid read of size 32
==167396==    at 0x78971D69: mkl_dft_avx2_gather_z_z (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x78D098DA: _row_1d_dft_twist (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x78D08263: mkl_dft_avx2_xz_4step_1_2 (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x62190337: step1234 (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62191154: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x63B35282: __kmp_invoke_microtask (in /opt/intel/oneapi/compiler/2024.1/lib/libiomp5.so)
==167396==    by 0x63AC3972: __kmp_invoke_task_func (kmp_runtime.cpp:8487)
==167396==    by 0x63ABCDA9: __kmp_fork_call (kmp_runtime.cpp:2755)
==167396==    by 0x63A7E3C7: __kmpc_fork_call (kmp_csupport.cpp:379)
==167396==    by 0x62191060: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62190921: mkl_dft_z1d_via_2d_f (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x7895FD61: mkl_dft_avx2_compute_fwd_d_c2c_1d_o (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==  Address 0x57aab680 is 0 bytes after a block of size 32,768 alloc'd
==167396==    at 0x4849013: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==167396==    by 0x5BCBBC8E: allocate (new_allocator.h:137)
==167396==    by 0x5BCBBC8E: allocate (alloc_traits.h:464)
==167396==    by 0x5BCBBC8E: _M_allocate (stl_vector.h:378)
==167396==    by 0x5BCBBC8E: _M_create_storage (stl_vector.h:395)
==167396==    by 0x5BCBBC8E: _Vector_base (stl_vector.h:332)
==167396==    by 0x5BCBBC8E: vector (stl_vector.h:597)
==167396==    by 0x5BCBBC8E: __pyx_pf_10custom_fft_4MKL_forward_fft_R2C (fft.cpp:3671)
==167396==    by 0x5BCBBC8E: __pyx_pw_10custom_fft_5MKL_forward_fft_R2C(_object*, _object* const*, long, _object*) (fft.cpp:3629)
==167396==    by 0x24B26C: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==    by 0x2479C5: ??? (in /usr/bin/python3.10)
==167396==    by 0x33D255: PyEval_EvalCode (in /usr/bin/python3.10)
==167396==    by 0x368107: ??? (in /usr/bin/python3.10)
==167396==    by 0x3619CA: ??? (in /usr/bin/python3.10)
==167396==    by 0x367E54: ??? (in /usr/bin/python3.10)
==167396==    by 0x367337: _PyRun_SimpleFileObject (in /usr/bin/python3.10)
==167396==    by 0x366F82: _PyRun_AnyFileObject (in /usr/bin/python3.10)
==167396==    by 0x359A5D: Py_RunMain (in /usr/bin/python3.10)
==167396==    by 0x33002C: Py_BytesMain (in /usr/bin/python3.10)
==167396== 
==167396== Invalid read of size 32
==167396==    at 0x78971D72: mkl_dft_avx2_gather_z_z (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x78D098DA: _row_1d_dft_twist (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x78D08263: mkl_dft_avx2_xz_4step_1_2 (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x62190337: step1234 (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62191154: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x63B35282: __kmp_invoke_microtask (in /opt/intel/oneapi/compiler/2024.1/lib/libiomp5.so)
==167396==    by 0x63AC3972: __kmp_invoke_task_func (kmp_runtime.cpp:8487)
==167396==    by 0x63ABCDA9: __kmp_fork_call (kmp_runtime.cpp:2755)
==167396==    by 0x63A7E3C7: __kmpc_fork_call (kmp_csupport.cpp:379)
==167396==    by 0x62191060: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62190921: mkl_dft_z1d_via_2d_f (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x7895FD61: mkl_dft_avx2_compute_fwd_d_c2c_1d_o (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==  Address 0x57aab6a0 is 32 bytes before an unallocated block of size 11,040 in arena "client"
==167396== 
==167396== Invalid read of size 32
==167396==    at 0x78971D78: mkl_dft_avx2_gather_z_z (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x78D098DA: _row_1d_dft_twist (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x78D08263: mkl_dft_avx2_xz_4step_1_2 (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x62190337: step1234 (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62191154: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x63B35282: __kmp_invoke_microtask (in /opt/intel/oneapi/compiler/2024.1/lib/libiomp5.so)
==167396==    by 0x63AC3972: __kmp_invoke_task_func (kmp_runtime.cpp:8487)
==167396==    by 0x63ABCDA9: __kmp_fork_call (kmp_runtime.cpp:2755)
==167396==    by 0x63A7E3C7: __kmpc_fork_call (kmp_csupport.cpp:379)
==167396==    by 0x62191060: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62190921: mkl_dft_z1d_via_2d_f (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x7895FD61: mkl_dft_avx2_compute_fwd_d_c2c_1d_o (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==  Address 0x57aaba80 is 960 bytes inside an unallocated block of size 11,040 in arena "client"
==167396== 
==167396== Invalid read of size 32
==167396==    at 0x78971D7E: mkl_dft_avx2_gather_z_z (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x78D098DA: _row_1d_dft_twist (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x78D08263: mkl_dft_avx2_xz_4step_1_2 (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x62190337: step1234 (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62191154: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x63B35282: __kmp_invoke_microtask (in /opt/intel/oneapi/compiler/2024.1/lib/libiomp5.so)
==167396==    by 0x63AC3972: __kmp_invoke_task_func (kmp_runtime.cpp:8487)
==167396==    by 0x63ABCDA9: __kmp_fork_call (kmp_runtime.cpp:2755)
==167396==    by 0x63A7E3C7: __kmpc_fork_call (kmp_csupport.cpp:379)
==167396==    by 0x62191060: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62190921: mkl_dft_z1d_via_2d_f (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x7895FD61: mkl_dft_avx2_compute_fwd_d_c2c_1d_o (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==  Address 0x57aabaa0 is 992 bytes inside an unallocated block of size 11,040 in arena "client"
==167396== 
==167396== Invalid read of size 32
==167396==    at 0x78971D85: mkl_dft_avx2_gather_z_z (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x78D098DA: _row_1d_dft_twist (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x78D08263: mkl_dft_avx2_xz_4step_1_2 (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x62190337: step1234 (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62191154: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x63B35282: __kmp_invoke_microtask (in /opt/intel/oneapi/compiler/2024.1/lib/libiomp5.so)
==167396==    by 0x63AC3972: __kmp_invoke_task_func (kmp_runtime.cpp:8487)
==167396==    by 0x63ABCDA9: __kmp_fork_call (kmp_runtime.cpp:2755)
==167396==    by 0x63A7E3C7: __kmpc_fork_call (kmp_csupport.cpp:379)
==167396==    by 0x62191060: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62190921: mkl_dft_z1d_via_2d_f (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x7895FD61: mkl_dft_avx2_compute_fwd_d_c2c_1d_o (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==  Address 0x57aabe80 is 1,984 bytes inside an unallocated block of size 11,040 in arena "client"
==167396== 
==167396== Invalid read of size 32
==167396==    at 0x78971D8A: mkl_dft_avx2_gather_z_z (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x78D098DA: _row_1d_dft_twist (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x78D08263: mkl_dft_avx2_xz_4step_1_2 (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x62190337: step1234 (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62191154: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x63B35282: __kmp_invoke_microtask (in /opt/intel/oneapi/compiler/2024.1/lib/libiomp5.so)
==167396==    by 0x63AC3972: __kmp_invoke_task_func (kmp_runtime.cpp:8487)
==167396==    by 0x63ABCDA9: __kmp_fork_call (kmp_runtime.cpp:2755)
==167396==    by 0x63A7E3C7: __kmpc_fork_call (kmp_csupport.cpp:379)
==167396==    by 0x62191060: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62190921: mkl_dft_z1d_via_2d_f (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x7895FD61: mkl_dft_avx2_compute_fwd_d_c2c_1d_o (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==  Address 0x57aabea0 is 2,016 bytes inside an unallocated block of size 11,040 in arena "client"
==167396== 
==167396== Invalid read of size 32
==167396==    at 0x78971D90: mkl_dft_avx2_gather_z_z (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x78D098DA: _row_1d_dft_twist (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x78D08263: mkl_dft_avx2_xz_4step_1_2 (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x62190337: step1234 (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62191154: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x63B35282: __kmp_invoke_microtask (in /opt/intel/oneapi/compiler/2024.1/lib/libiomp5.so)
==167396==    by 0x63AC3972: __kmp_invoke_task_func (kmp_runtime.cpp:8487)
==167396==    by 0x63ABCDA9: __kmp_fork_call (kmp_runtime.cpp:2755)
==167396==    by 0x63A7E3C7: __kmpc_fork_call (kmp_csupport.cpp:379)
==167396==    by 0x62191060: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62190921: mkl_dft_z1d_via_2d_f (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x7895FD61: mkl_dft_avx2_compute_fwd_d_c2c_1d_o (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==  Address 0x57aac280 is 3,008 bytes inside an unallocated block of size 11,040 in arena "client"
==167396== 
==167396== Invalid read of size 32
==167396==    at 0x78971D95: mkl_dft_avx2_gather_z_z (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x78D098DA: _row_1d_dft_twist (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x78D08263: mkl_dft_avx2_xz_4step_1_2 (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x62190337: step1234 (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62191154: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x63B35282: __kmp_invoke_microtask (in /opt/intel/oneapi/compiler/2024.1/lib/libiomp5.so)
==167396==    by 0x63AC3972: __kmp_invoke_task_func (kmp_runtime.cpp:8487)
==167396==    by 0x63ABCDA9: __kmp_fork_call (kmp_runtime.cpp:2755)
==167396==    by 0x63A7E3C7: __kmpc_fork_call (kmp_csupport.cpp:379)
==167396==    by 0x62191060: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62190921: mkl_dft_z1d_via_2d_f (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x7895FD61: mkl_dft_avx2_compute_fwd_d_c2c_1d_o (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==  Address 0x57aac2a0 is 3,040 bytes inside an unallocated block of size 11,040 in arena "client"
==167396== 
==167396== Invalid write of size 8
==167396==    at 0x789813EA: mkl_dft_avx2_scatter_z_z (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x621906A7: step1234 (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62191154: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x63B35282: __kmp_invoke_microtask (in /opt/intel/oneapi/compiler/2024.1/lib/libiomp5.so)
==167396==    by 0x63AC3972: __kmp_invoke_task_func (kmp_runtime.cpp:8487)
==167396==    by 0x63ABCDA9: __kmp_fork_call (kmp_runtime.cpp:2755)
==167396==    by 0x63A7E3C7: __kmpc_fork_call (kmp_csupport.cpp:379)
==167396==    by 0x62191060: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62190921: mkl_dft_z1d_via_2d_f (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x7895FD61: mkl_dft_avx2_compute_fwd_d_c2c_1d_o (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x78775EA3: compute_r2c_1d_even_fwd (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x5C8C63E9: DftiComputeForward (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_lp64.so.2)
==167396==  Address 0x57b495a8 is 0 bytes after a block of size 32,776 alloc'd
==167396==    at 0x4849013: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==167396==    by 0x5BCBD954: allocate (new_allocator.h:137)
==167396==    by 0x5BCBD954: allocate (alloc_traits.h:464)
==167396==    by 0x5BCBD954: _M_allocate (stl_vector.h:378)
==167396==    by 0x5BCBD954: _M_create_storage (stl_vector.h:395)
==167396==    by 0x5BCBD954: _Vector_base (stl_vector.h:332)
==167396==    by 0x5BCBD954: vector (stl_vector.h:552)
==167396==    by 0x5BCBD954: _forward_fft_R2C(std::vector<float, std::allocator<float> >) (mkl_fft.cpp:22)
==167396==    by 0x5BCBBCD3: __pyx_pf_10custom_fft_4MKL_forward_fft_R2C (fft.cpp:3671)
==167396==    by 0x5BCBBCD3: __pyx_pw_10custom_fft_5MKL_forward_fft_R2C(_object*, _object* const*, long, _object*) (fft.cpp:3629)
==167396==    by 0x24B26C: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==    by 0x2479C5: ??? (in /usr/bin/python3.10)
==167396==    by 0x33D255: PyEval_EvalCode (in /usr/bin/python3.10)
==167396==    by 0x368107: ??? (in /usr/bin/python3.10)
==167396==    by 0x3619CA: ??? (in /usr/bin/python3.10)
==167396==    by 0x367E54: ??? (in /usr/bin/python3.10)
==167396==    by 0x367337: _PyRun_SimpleFileObject (in /usr/bin/python3.10)
==167396==    by 0x366F82: _PyRun_AnyFileObject (in /usr/bin/python3.10)
==167396==    by 0x359A5D: Py_RunMain (in /usr/bin/python3.10)
==167396== 
==167396== Invalid write of size 8
==167396==    at 0x789813EE: mkl_dft_avx2_scatter_z_z (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x621906A7: step1234 (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62191154: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x63B35282: __kmp_invoke_microtask (in /opt/intel/oneapi/compiler/2024.1/lib/libiomp5.so)
==167396==    by 0x63AC3972: __kmp_invoke_task_func (kmp_runtime.cpp:8487)
==167396==    by 0x63ABCDA9: __kmp_fork_call (kmp_runtime.cpp:2755)
==167396==    by 0x63A7E3C7: __kmpc_fork_call (kmp_csupport.cpp:379)
==167396==    by 0x62191060: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62190921: mkl_dft_z1d_via_2d_f (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x7895FD61: mkl_dft_avx2_compute_fwd_d_c2c_1d_o (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x78775EA3: compute_r2c_1d_even_fwd (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x5C8C63E9: DftiComputeForward (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_lp64.so.2)
==167396==  Address 0x57b495c0 is 16 bytes after a block of size 32,784 in arena "client"
==167396== 
==167396== Invalid write of size 8
==167396==    at 0x7898147A: mkl_dft_avx2_scatter_z_z (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x621906A7: step1234 (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62191154: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x63B35282: __kmp_invoke_microtask (in /opt/intel/oneapi/compiler/2024.1/lib/libiomp5.so)
==167396==    by 0x63AC3972: __kmp_invoke_task_func (kmp_runtime.cpp:8487)
==167396==    by 0x63ABCDA9: __kmp_fork_call (kmp_runtime.cpp:2755)
==167396==    by 0x63A7E3C7: __kmpc_fork_call (kmp_csupport.cpp:379)
==167396==    by 0x62191060: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62190921: mkl_dft_z1d_via_2d_f (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x7895FD61: mkl_dft_avx2_compute_fwd_d_c2c_1d_o (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x78775EA3: compute_r2c_1d_even_fwd (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x5C8C63E9: DftiComputeForward (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_lp64.so.2)
==167396==  Address 0x57b495e0 is 16 bytes before a block of size 8,480 alloc'd
==167396==    at 0x4848899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==167396==    by 0x5D487B7A: mkl_serv_allocate (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_core.so.2)
==167396==    by 0x78D08203: mkl_dft_avx2_xz_4step_1_2 (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x62190337: step1234 (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62191154: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x63B35282: __kmp_invoke_microtask (in /opt/intel/oneapi/compiler/2024.1/lib/libiomp5.so)
==167396==    by 0x63AC3972: __kmp_invoke_task_func (kmp_runtime.cpp:8487)
==167396==    by 0x63ABCDA9: __kmp_fork_call (kmp_runtime.cpp:2755)
==167396==    by 0x63A7E3C7: __kmpc_fork_call (kmp_csupport.cpp:379)
==167396==    by 0x62191060: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62190921: mkl_dft_z1d_via_2d_f (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x7895FD61: mkl_dft_avx2_compute_fwd_d_c2c_1d_o (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396== 
==167396== Invalid write of size 8
==167396==    at 0x789813FE: mkl_dft_avx2_scatter_z_z (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x621906A7: step1234 (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62191154: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x63B35282: __kmp_invoke_microtask (in /opt/intel/oneapi/compiler/2024.1/lib/libiomp5.so)
==167396==    by 0x63AC3972: __kmp_invoke_task_func (kmp_runtime.cpp:8487)
==167396==    by 0x63ABCDA9: __kmp_fork_call (kmp_runtime.cpp:2755)
==167396==    by 0x63A7E3C7: __kmpc_fork_call (kmp_csupport.cpp:379)
==167396==    by 0x62191060: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62190921: mkl_dft_z1d_via_2d_f (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x7895FD61: mkl_dft_avx2_compute_fwd_d_c2c_1d_o (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x78775EA3: compute_r2c_1d_even_fwd (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x5C8C63E9: DftiComputeForward (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_lp64.so.2)
==167396==  Address 0x57b4dda0 is 272 bytes inside a block of size 1,481 free'd
==167396==    at 0x484DCD3: realloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==167396==    by 0x2694E1: ??? (in /usr/bin/python3.10)
==167396==    by 0x268D40: _PyUnicodeWriter_PrepareInternal (in /usr/bin/python3.10)
==167396==    by 0x272DED: ??? (in /usr/bin/python3.10)
==167396==    by 0x26210D: ??? (in /usr/bin/python3.10)
==167396==    by 0x258A7A: _PyObject_MakeTpCall (in /usr/bin/python3.10)
==167396==    by 0x25214F: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==    by 0x2479C5: ??? (in /usr/bin/python3.10)
==167396==    by 0x33D255: PyEval_EvalCode (in /usr/bin/python3.10)
==167396==    by 0x342E2C: ??? (in /usr/bin/python3.10)
==167396==    by 0x262C58: ??? (in /usr/bin/python3.10)
==167396==    by 0x24D5D6: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==  Block was alloc'd at
==167396==    at 0x4848899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==167396==    by 0x268E48: _PyUnicodeWriter_PrepareInternal (in /usr/bin/python3.10)
==167396==    by 0x272DED: ??? (in /usr/bin/python3.10)
==167396==    by 0x26210D: ??? (in /usr/bin/python3.10)
==167396==    by 0x258A7A: _PyObject_MakeTpCall (in /usr/bin/python3.10)
==167396==    by 0x25214F: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==    by 0x2479C5: ??? (in /usr/bin/python3.10)
==167396==    by 0x33D255: PyEval_EvalCode (in /usr/bin/python3.10)
==167396==    by 0x342E2C: ??? (in /usr/bin/python3.10)
==167396==    by 0x262C58: ??? (in /usr/bin/python3.10)
==167396==    by 0x24D5D6: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==    by 0x2629FB: _PyFunction_Vectorcall (in /usr/bin/python3.10)
==167396== 
==167396== Invalid write of size 8
==167396==    at 0x78981404: mkl_dft_avx2_scatter_z_z (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x621906A7: step1234 (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62191154: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x63B35282: __kmp_invoke_microtask (in /opt/intel/oneapi/compiler/2024.1/lib/libiomp5.so)
==167396==    by 0x63AC3972: __kmp_invoke_task_func (kmp_runtime.cpp:8487)
==167396==    by 0x63ABCDA9: __kmp_fork_call (kmp_runtime.cpp:2755)
==167396==    by 0x63A7E3C7: __kmpc_fork_call (kmp_csupport.cpp:379)
==167396==    by 0x62191060: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62190921: mkl_dft_z1d_via_2d_f (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x7895FD61: mkl_dft_avx2_compute_fwd_d_c2c_1d_o (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x78775EA3: compute_r2c_1d_even_fwd (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x5C8C63E9: DftiComputeForward (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_lp64.so.2)
==167396==  Address 0x57b4ddc0 is 304 bytes inside a block of size 1,481 free'd
==167396==    at 0x484DCD3: realloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==167396==    by 0x2694E1: ??? (in /usr/bin/python3.10)
==167396==    by 0x268D40: _PyUnicodeWriter_PrepareInternal (in /usr/bin/python3.10)
==167396==    by 0x272DED: ??? (in /usr/bin/python3.10)
==167396==    by 0x26210D: ??? (in /usr/bin/python3.10)
==167396==    by 0x258A7A: _PyObject_MakeTpCall (in /usr/bin/python3.10)
==167396==    by 0x25214F: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==    by 0x2479C5: ??? (in /usr/bin/python3.10)
==167396==    by 0x33D255: PyEval_EvalCode (in /usr/bin/python3.10)
==167396==    by 0x342E2C: ??? (in /usr/bin/python3.10)
==167396==    by 0x262C58: ??? (in /usr/bin/python3.10)
==167396==    by 0x24D5D6: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==  Block was alloc'd at
==167396==    at 0x4848899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==167396==    by 0x268E48: _PyUnicodeWriter_PrepareInternal (in /usr/bin/python3.10)
==167396==    by 0x272DED: ??? (in /usr/bin/python3.10)
==167396==    by 0x26210D: ??? (in /usr/bin/python3.10)
==167396==    by 0x258A7A: _PyObject_MakeTpCall (in /usr/bin/python3.10)
==167396==    by 0x25214F: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==    by 0x2479C5: ??? (in /usr/bin/python3.10)
==167396==    by 0x33D255: PyEval_EvalCode (in /usr/bin/python3.10)
==167396==    by 0x342E2C: ??? (in /usr/bin/python3.10)
==167396==    by 0x262C58: ??? (in /usr/bin/python3.10)
==167396==    by 0x24D5D6: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==    by 0x2629FB: _PyFunction_Vectorcall (in /usr/bin/python3.10)
==167396== 
==167396== Invalid write of size 8
==167396==    at 0x7898140B: mkl_dft_avx2_scatter_z_z (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x621906A7: step1234 (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62191154: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x63B35282: __kmp_invoke_microtask (in /opt/intel/oneapi/compiler/2024.1/lib/libiomp5.so)
==167396==    by 0x63AC3972: __kmp_invoke_task_func (kmp_runtime.cpp:8487)
==167396==    by 0x63ABCDA9: __kmp_fork_call (kmp_runtime.cpp:2755)
==167396==    by 0x63A7E3C7: __kmpc_fork_call (kmp_csupport.cpp:379)
==167396==    by 0x62191060: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62190921: mkl_dft_z1d_via_2d_f (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x7895FD61: mkl_dft_avx2_compute_fwd_d_c2c_1d_o (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x78775EA3: compute_r2c_1d_even_fwd (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x5C8C63E9: DftiComputeForward (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_lp64.so.2)
==167396==  Address 0x57b4e1a0 is 1,296 bytes inside a block of size 1,481 free'd
==167396==    at 0x484DCD3: realloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==167396==    by 0x2694E1: ??? (in /usr/bin/python3.10)
==167396==    by 0x268D40: _PyUnicodeWriter_PrepareInternal (in /usr/bin/python3.10)
==167396==    by 0x272DED: ??? (in /usr/bin/python3.10)
==167396==    by 0x26210D: ??? (in /usr/bin/python3.10)
==167396==    by 0x258A7A: _PyObject_MakeTpCall (in /usr/bin/python3.10)
==167396==    by 0x25214F: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==    by 0x2479C5: ??? (in /usr/bin/python3.10)
==167396==    by 0x33D255: PyEval_EvalCode (in /usr/bin/python3.10)
==167396==    by 0x342E2C: ??? (in /usr/bin/python3.10)
==167396==    by 0x262C58: ??? (in /usr/bin/python3.10)
==167396==    by 0x24D5D6: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==  Block was alloc'd at
==167396==    at 0x4848899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==167396==    by 0x268E48: _PyUnicodeWriter_PrepareInternal (in /usr/bin/python3.10)
==167396==    by 0x272DED: ??? (in /usr/bin/python3.10)
==167396==    by 0x26210D: ??? (in /usr/bin/python3.10)
==167396==    by 0x258A7A: _PyObject_MakeTpCall (in /usr/bin/python3.10)
==167396==    by 0x25214F: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==    by 0x2479C5: ??? (in /usr/bin/python3.10)
==167396==    by 0x33D255: PyEval_EvalCode (in /usr/bin/python3.10)
==167396==    by 0x342E2C: ??? (in /usr/bin/python3.10)
==167396==    by 0x262C58: ??? (in /usr/bin/python3.10)
==167396==    by 0x24D5D6: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==    by 0x2629FB: _PyFunction_Vectorcall (in /usr/bin/python3.10)
==167396== 
==167396== Invalid write of size 8
==167396==    at 0x78981411: mkl_dft_avx2_scatter_z_z (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x621906A7: step1234 (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62191154: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x63B35282: __kmp_invoke_microtask (in /opt/intel/oneapi/compiler/2024.1/lib/libiomp5.so)
==167396==    by 0x63AC3972: __kmp_invoke_task_func (kmp_runtime.cpp:8487)
==167396==    by 0x63ABCDA9: __kmp_fork_call (kmp_runtime.cpp:2755)
==167396==    by 0x63A7E3C7: __kmpc_fork_call (kmp_csupport.cpp:379)
==167396==    by 0x62191060: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62190921: mkl_dft_z1d_via_2d_f (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x7895FD61: mkl_dft_avx2_compute_fwd_d_c2c_1d_o (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x78775EA3: compute_r2c_1d_even_fwd (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x5C8C63E9: DftiComputeForward (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_lp64.so.2)
==167396==  Address 0x57b4e1c0 is 1,328 bytes inside a block of size 1,481 free'd
==167396==    at 0x484DCD3: realloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==167396==    by 0x2694E1: ??? (in /usr/bin/python3.10)
==167396==    by 0x268D40: _PyUnicodeWriter_PrepareInternal (in /usr/bin/python3.10)
==167396==    by 0x272DED: ??? (in /usr/bin/python3.10)
==167396==    by 0x26210D: ??? (in /usr/bin/python3.10)
==167396==    by 0x258A7A: _PyObject_MakeTpCall (in /usr/bin/python3.10)
==167396==    by 0x25214F: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==    by 0x2479C5: ??? (in /usr/bin/python3.10)
==167396==    by 0x33D255: PyEval_EvalCode (in /usr/bin/python3.10)
==167396==    by 0x342E2C: ??? (in /usr/bin/python3.10)
==167396==    by 0x262C58: ??? (in /usr/bin/python3.10)
==167396==    by 0x24D5D6: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==  Block was alloc'd at
==167396==    at 0x4848899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==167396==    by 0x268E48: _PyUnicodeWriter_PrepareInternal (in /usr/bin/python3.10)
==167396==    by 0x272DED: ??? (in /usr/bin/python3.10)
==167396==    by 0x26210D: ??? (in /usr/bin/python3.10)
==167396==    by 0x258A7A: _PyObject_MakeTpCall (in /usr/bin/python3.10)
==167396==    by 0x25214F: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==    by 0x2479C5: ??? (in /usr/bin/python3.10)
==167396==    by 0x33D255: PyEval_EvalCode (in /usr/bin/python3.10)
==167396==    by 0x342E2C: ??? (in /usr/bin/python3.10)
==167396==    by 0x262C58: ??? (in /usr/bin/python3.10)
==167396==    by 0x24D5D6: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==    by 0x2629FB: _PyFunction_Vectorcall (in /usr/bin/python3.10)
==167396== 
==167396== Invalid write of size 8
==167396==    at 0x78981490: mkl_dft_avx2_scatter_z_z (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x621906A7: step1234 (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62191154: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x63B35282: __kmp_invoke_microtask (in /opt/intel/oneapi/compiler/2024.1/lib/libiomp5.so)
==167396==    by 0x63AC3972: __kmp_invoke_task_func (kmp_runtime.cpp:8487)
==167396==    by 0x63ABCDA9: __kmp_fork_call (kmp_runtime.cpp:2755)
==167396==    by 0x63A7E3C7: __kmpc_fork_call (kmp_csupport.cpp:379)
==167396==    by 0x62191060: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62190921: mkl_dft_z1d_via_2d_f (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x7895FD61: mkl_dft_avx2_compute_fwd_d_c2c_1d_o (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x78775EA3: compute_r2c_1d_even_fwd (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x5C8C63E9: DftiComputeForward (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_lp64.so.2)
==167396==  Address 0x57b4dde0 is 336 bytes inside a block of size 1,481 free'd
==167396==    at 0x484DCD3: realloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==167396==    by 0x2694E1: ??? (in /usr/bin/python3.10)
==167396==    by 0x268D40: _PyUnicodeWriter_PrepareInternal (in /usr/bin/python3.10)
==167396==    by 0x272DED: ??? (in /usr/bin/python3.10)
==167396==    by 0x26210D: ??? (in /usr/bin/python3.10)
==167396==    by 0x258A7A: _PyObject_MakeTpCall (in /usr/bin/python3.10)
==167396==    by 0x25214F: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==    by 0x2479C5: ??? (in /usr/bin/python3.10)
==167396==    by 0x33D255: PyEval_EvalCode (in /usr/bin/python3.10)
==167396==    by 0x342E2C: ??? (in /usr/bin/python3.10)
==167396==    by 0x262C58: ??? (in /usr/bin/python3.10)
==167396==    by 0x24D5D6: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==  Block was alloc'd at
==167396==    at 0x4848899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==167396==    by 0x268E48: _PyUnicodeWriter_PrepareInternal (in /usr/bin/python3.10)
==167396==    by 0x272DED: ??? (in /usr/bin/python3.10)
==167396==    by 0x26210D: ??? (in /usr/bin/python3.10)
==167396==    by 0x258A7A: _PyObject_MakeTpCall (in /usr/bin/python3.10)
==167396==    by 0x25214F: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==    by 0x2479C5: ??? (in /usr/bin/python3.10)
==167396==    by 0x33D255: PyEval_EvalCode (in /usr/bin/python3.10)
==167396==    by 0x342E2C: ??? (in /usr/bin/python3.10)
==167396==    by 0x262C58: ??? (in /usr/bin/python3.10)
==167396==    by 0x24D5D6: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==    by 0x2629FB: _PyFunction_Vectorcall (in /usr/bin/python3.10)
==167396== 
==167396== Invalid write of size 8
==167396==    at 0x78981497: mkl_dft_avx2_scatter_z_z (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x621906A7: step1234 (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62191154: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x63B35282: __kmp_invoke_microtask (in /opt/intel/oneapi/compiler/2024.1/lib/libiomp5.so)
==167396==    by 0x63AC3972: __kmp_invoke_task_func (kmp_runtime.cpp:8487)
==167396==    by 0x63ABCDA9: __kmp_fork_call (kmp_runtime.cpp:2755)
==167396==    by 0x63A7E3C7: __kmpc_fork_call (kmp_csupport.cpp:379)
==167396==    by 0x62191060: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62190921: mkl_dft_z1d_via_2d_f (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x7895FD61: mkl_dft_avx2_compute_fwd_d_c2c_1d_o (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x78775EA3: compute_r2c_1d_even_fwd (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x5C8C63E9: DftiComputeForward (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_lp64.so.2)
==167396==  Address 0x57b4de00 is 368 bytes inside a block of size 1,481 free'd
==167396==    at 0x484DCD3: realloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==167396==    by 0x2694E1: ??? (in /usr/bin/python3.10)
==167396==    by 0x268D40: _PyUnicodeWriter_PrepareInternal (in /usr/bin/python3.10)
==167396==    by 0x272DED: ??? (in /usr/bin/python3.10)
==167396==    by 0x26210D: ??? (in /usr/bin/python3.10)
==167396==    by 0x258A7A: _PyObject_MakeTpCall (in /usr/bin/python3.10)
==167396==    by 0x25214F: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==    by 0x2479C5: ??? (in /usr/bin/python3.10)
==167396==    by 0x33D255: PyEval_EvalCode (in /usr/bin/python3.10)
==167396==    by 0x342E2C: ??? (in /usr/bin/python3.10)
==167396==    by 0x262C58: ??? (in /usr/bin/python3.10)
==167396==    by 0x24D5D6: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==  Block was alloc'd at
==167396==    at 0x4848899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==167396==    by 0x268E48: _PyUnicodeWriter_PrepareInternal (in /usr/bin/python3.10)
==167396==    by 0x272DED: ??? (in /usr/bin/python3.10)
==167396==    by 0x26210D: ??? (in /usr/bin/python3.10)
==167396==    by 0x258A7A: _PyObject_MakeTpCall (in /usr/bin/python3.10)
==167396==    by 0x25214F: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==    by 0x2479C5: ??? (in /usr/bin/python3.10)
==167396==    by 0x33D255: PyEval_EvalCode (in /usr/bin/python3.10)
==167396==    by 0x342E2C: ??? (in /usr/bin/python3.10)
==167396==    by 0x262C58: ??? (in /usr/bin/python3.10)
==167396==    by 0x24D5D6: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==    by 0x2629FB: _PyFunction_Vectorcall (in /usr/bin/python3.10)
==167396== 
==167396== Invalid write of size 8
==167396==    at 0x7898149E: mkl_dft_avx2_scatter_z_z (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x621906A7: step1234 (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62191154: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x63B35282: __kmp_invoke_microtask (in /opt/intel/oneapi/compiler/2024.1/lib/libiomp5.so)
==167396==    by 0x63AC3972: __kmp_invoke_task_func (kmp_runtime.cpp:8487)
==167396==    by 0x63ABCDA9: __kmp_fork_call (kmp_runtime.cpp:2755)
==167396==    by 0x63A7E3C7: __kmpc_fork_call (kmp_csupport.cpp:379)
==167396==    by 0x62191060: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62190921: mkl_dft_z1d_via_2d_f (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x7895FD61: mkl_dft_avx2_compute_fwd_d_c2c_1d_o (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x78775EA3: compute_r2c_1d_even_fwd (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x5C8C63E9: DftiComputeForward (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_lp64.so.2)
==167396==  Address 0x57b4e1e0 is 1,360 bytes inside a block of size 1,481 free'd
==167396==    at 0x484DCD3: realloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==167396==    by 0x2694E1: ??? (in /usr/bin/python3.10)
==167396==    by 0x268D40: _PyUnicodeWriter_PrepareInternal (in /usr/bin/python3.10)
==167396==    by 0x272DED: ??? (in /usr/bin/python3.10)
==167396==    by 0x26210D: ??? (in /usr/bin/python3.10)
==167396==    by 0x258A7A: _PyObject_MakeTpCall (in /usr/bin/python3.10)
==167396==    by 0x25214F: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==    by 0x2479C5: ??? (in /usr/bin/python3.10)
==167396==    by 0x33D255: PyEval_EvalCode (in /usr/bin/python3.10)
==167396==    by 0x342E2C: ??? (in /usr/bin/python3.10)
==167396==    by 0x262C58: ??? (in /usr/bin/python3.10)
==167396==    by 0x24D5D6: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==  Block was alloc'd at
==167396==    at 0x4848899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==167396==    by 0x268E48: _PyUnicodeWriter_PrepareInternal (in /usr/bin/python3.10)
==167396==    by 0x272DED: ??? (in /usr/bin/python3.10)
==167396==    by 0x26210D: ??? (in /usr/bin/python3.10)
==167396==    by 0x258A7A: _PyObject_MakeTpCall (in /usr/bin/python3.10)
==167396==    by 0x25214F: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==    by 0x2479C5: ??? (in /usr/bin/python3.10)
==167396==    by 0x33D255: PyEval_EvalCode (in /usr/bin/python3.10)
==167396==    by 0x342E2C: ??? (in /usr/bin/python3.10)
==167396==    by 0x262C58: ??? (in /usr/bin/python3.10)
==167396==    by 0x24D5D6: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==    by 0x2629FB: _PyFunction_Vectorcall (in /usr/bin/python3.10)
==167396== 
==167396== Invalid write of size 8
==167396==    at 0x789814A5: mkl_dft_avx2_scatter_z_z (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x621906A7: step1234 (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62191154: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x63B35282: __kmp_invoke_microtask (in /opt/intel/oneapi/compiler/2024.1/lib/libiomp5.so)
==167396==    by 0x63AC3972: __kmp_invoke_task_func (kmp_runtime.cpp:8487)
==167396==    by 0x63ABCDA9: __kmp_fork_call (kmp_runtime.cpp:2755)
==167396==    by 0x63A7E3C7: __kmpc_fork_call (kmp_csupport.cpp:379)
==167396==    by 0x62191060: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62190921: mkl_dft_z1d_via_2d_f (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x7895FD61: mkl_dft_avx2_compute_fwd_d_c2c_1d_o (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x78775EA3: compute_r2c_1d_even_fwd (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x5C8C63E9: DftiComputeForward (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_lp64.so.2)
==167396==  Address 0x57b4e200 is 1,392 bytes inside a block of size 1,481 free'd
==167396==    at 0x484DCD3: realloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==167396==    by 0x2694E1: ??? (in /usr/bin/python3.10)
==167396==    by 0x268D40: _PyUnicodeWriter_PrepareInternal (in /usr/bin/python3.10)
==167396==    by 0x272DED: ??? (in /usr/bin/python3.10)
==167396==    by 0x26210D: ??? (in /usr/bin/python3.10)
==167396==    by 0x258A7A: _PyObject_MakeTpCall (in /usr/bin/python3.10)
==167396==    by 0x25214F: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==    by 0x2479C5: ??? (in /usr/bin/python3.10)
==167396==    by 0x33D255: PyEval_EvalCode (in /usr/bin/python3.10)
==167396==    by 0x342E2C: ??? (in /usr/bin/python3.10)
==167396==    by 0x262C58: ??? (in /usr/bin/python3.10)
==167396==    by 0x24D5D6: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==  Block was alloc'd at
==167396==    at 0x4848899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==167396==    by 0x268E48: _PyUnicodeWriter_PrepareInternal (in /usr/bin/python3.10)
==167396==    by 0x272DED: ??? (in /usr/bin/python3.10)
==167396==    by 0x26210D: ??? (in /usr/bin/python3.10)
==167396==    by 0x258A7A: _PyObject_MakeTpCall (in /usr/bin/python3.10)
==167396==    by 0x25214F: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==    by 0x2479C5: ??? (in /usr/bin/python3.10)
==167396==    by 0x33D255: PyEval_EvalCode (in /usr/bin/python3.10)
==167396==    by 0x342E2C: ??? (in /usr/bin/python3.10)
==167396==    by 0x262C58: ??? (in /usr/bin/python3.10)
==167396==    by 0x24D5D6: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==    by 0x2629FB: _PyFunction_Vectorcall (in /usr/bin/python3.10)
==167396== 
==167396== Invalid write of size 8
==167396==    at 0x789813F3: mkl_dft_avx2_scatter_z_z (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x621906A7: step1234 (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62191154: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x63B35282: __kmp_invoke_microtask (in /opt/intel/oneapi/compiler/2024.1/lib/libiomp5.so)
==167396==    by 0x63AC3972: __kmp_invoke_task_func (kmp_runtime.cpp:8487)
==167396==    by 0x63ABCDA9: __kmp_fork_call (kmp_runtime.cpp:2755)
==167396==    by 0x63A7E3C7: __kmpc_fork_call (kmp_csupport.cpp:379)
==167396==    by 0x62191060: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62190921: mkl_dft_z1d_via_2d_f (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x7895FD61: mkl_dft_avx2_compute_fwd_d_c2c_1d_o (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x78775EA3: compute_r2c_1d_even_fwd (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x5C8C63E9: DftiComputeForward (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_lp64.so.2)
==167396==  Address 0x57b4e9a0 is 1,792 bytes inside a block of size 2,141 free'd
==167396==    at 0x484DCD3: realloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==167396==    by 0x2694E1: ??? (in /usr/bin/python3.10)
==167396==    by 0x268D40: _PyUnicodeWriter_PrepareInternal (in /usr/bin/python3.10)
==167396==    by 0x275992: _PyUnicode_FormatAdvancedWriter (in /usr/bin/python3.10)
==167396==    by 0x27385D: ??? (in /usr/bin/python3.10)
==167396==    by 0x272CF2: ??? (in /usr/bin/python3.10)
==167396==    by 0x26210D: ??? (in /usr/bin/python3.10)
==167396==    by 0x258A7A: _PyObject_MakeTpCall (in /usr/bin/python3.10)
==167396==    by 0x25214F: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==    by 0x2479C5: ??? (in /usr/bin/python3.10)
==167396==    by 0x33D255: PyEval_EvalCode (in /usr/bin/python3.10)
==167396==    by 0x342E2C: ??? (in /usr/bin/python3.10)
==167396==  Block was alloc'd at
==167396==    at 0x484DCD3: realloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==167396==    by 0x2694E1: ??? (in /usr/bin/python3.10)
==167396==    by 0x268D40: _PyUnicodeWriter_PrepareInternal (in /usr/bin/python3.10)
==167396==    by 0x272DED: ??? (in /usr/bin/python3.10)
==167396==    by 0x26210D: ??? (in /usr/bin/python3.10)
==167396==    by 0x258A7A: _PyObject_MakeTpCall (in /usr/bin/python3.10)
==167396==    by 0x25214F: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==    by 0x2479C5: ??? (in /usr/bin/python3.10)
==167396==    by 0x33D255: PyEval_EvalCode (in /usr/bin/python3.10)
==167396==    by 0x342E2C: ??? (in /usr/bin/python3.10)
==167396==    by 0x262C58: ??? (in /usr/bin/python3.10)
==167396==    by 0x24D5D6: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396== 
==167396== Invalid write of size 8
==167396==    at 0x789813F8: mkl_dft_avx2_scatter_z_z (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x621906A7: step1234 (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62191154: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x63B35282: __kmp_invoke_microtask (in /opt/intel/oneapi/compiler/2024.1/lib/libiomp5.so)
==167396==    by 0x63AC3972: __kmp_invoke_task_func (kmp_runtime.cpp:8487)
==167396==    by 0x63ABCDA9: __kmp_fork_call (kmp_runtime.cpp:2755)
==167396==    by 0x63A7E3C7: __kmpc_fork_call (kmp_csupport.cpp:379)
==167396==    by 0x62191060: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62190921: mkl_dft_z1d_via_2d_f (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x7895FD61: mkl_dft_avx2_compute_fwd_d_c2c_1d_o (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x78775EA3: compute_r2c_1d_even_fwd (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x5C8C63E9: DftiComputeForward (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_lp64.so.2)
==167396==  Address 0x57b4e9c0 is 1,824 bytes inside a block of size 2,141 free'd
==167396==    at 0x484DCD3: realloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==167396==    by 0x2694E1: ??? (in /usr/bin/python3.10)
==167396==    by 0x268D40: _PyUnicodeWriter_PrepareInternal (in /usr/bin/python3.10)
==167396==    by 0x275992: _PyUnicode_FormatAdvancedWriter (in /usr/bin/python3.10)
==167396==    by 0x27385D: ??? (in /usr/bin/python3.10)
==167396==    by 0x272CF2: ??? (in /usr/bin/python3.10)
==167396==    by 0x26210D: ??? (in /usr/bin/python3.10)
==167396==    by 0x258A7A: _PyObject_MakeTpCall (in /usr/bin/python3.10)
==167396==    by 0x25214F: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==    by 0x2479C5: ??? (in /usr/bin/python3.10)
==167396==    by 0x33D255: PyEval_EvalCode (in /usr/bin/python3.10)
==167396==    by 0x342E2C: ??? (in /usr/bin/python3.10)
==167396==  Block was alloc'd at
==167396==    at 0x484DCD3: realloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==167396==    by 0x2694E1: ??? (in /usr/bin/python3.10)
==167396==    by 0x268D40: _PyUnicodeWriter_PrepareInternal (in /usr/bin/python3.10)
==167396==    by 0x272DED: ??? (in /usr/bin/python3.10)
==167396==    by 0x26210D: ??? (in /usr/bin/python3.10)
==167396==    by 0x258A7A: _PyObject_MakeTpCall (in /usr/bin/python3.10)
==167396==    by 0x25214F: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==    by 0x2479C5: ??? (in /usr/bin/python3.10)
==167396==    by 0x33D255: PyEval_EvalCode (in /usr/bin/python3.10)
==167396==    by 0x342E2C: ??? (in /usr/bin/python3.10)
==167396==    by 0x262C58: ??? (in /usr/bin/python3.10)
==167396==    by 0x24D5D6: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396== 
==167396== Invalid write of size 8
==167396==    at 0x7898147F: mkl_dft_avx2_scatter_z_z (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x621906A7: step1234 (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62191154: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x63B35282: __kmp_invoke_microtask (in /opt/intel/oneapi/compiler/2024.1/lib/libiomp5.so)
==167396==    by 0x63AC3972: __kmp_invoke_task_func (kmp_runtime.cpp:8487)
==167396==    by 0x63ABCDA9: __kmp_fork_call (kmp_runtime.cpp:2755)
==167396==    by 0x63A7E3C7: __kmpc_fork_call (kmp_csupport.cpp:379)
==167396==    by 0x62191060: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62190921: mkl_dft_z1d_via_2d_f (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x7895FD61: mkl_dft_avx2_compute_fwd_d_c2c_1d_o (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x78775EA3: compute_r2c_1d_even_fwd (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x5C8C63E9: DftiComputeForward (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_lp64.so.2)
==167396==  Address 0x57b4e600 is 864 bytes inside a block of size 2,141 free'd
==167396==    at 0x484DCD3: realloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==167396==    by 0x2694E1: ??? (in /usr/bin/python3.10)
==167396==    by 0x268D40: _PyUnicodeWriter_PrepareInternal (in /usr/bin/python3.10)
==167396==    by 0x275992: _PyUnicode_FormatAdvancedWriter (in /usr/bin/python3.10)
==167396==    by 0x27385D: ??? (in /usr/bin/python3.10)
==167396==    by 0x272CF2: ??? (in /usr/bin/python3.10)
==167396==    by 0x26210D: ??? (in /usr/bin/python3.10)
==167396==    by 0x258A7A: _PyObject_MakeTpCall (in /usr/bin/python3.10)
==167396==    by 0x25214F: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==    by 0x2479C5: ??? (in /usr/bin/python3.10)
==167396==    by 0x33D255: PyEval_EvalCode (in /usr/bin/python3.10)
==167396==    by 0x342E2C: ??? (in /usr/bin/python3.10)
==167396==  Block was alloc'd at
==167396==    at 0x484DCD3: realloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==167396==    by 0x2694E1: ??? (in /usr/bin/python3.10)
==167396==    by 0x268D40: _PyUnicodeWriter_PrepareInternal (in /usr/bin/python3.10)
==167396==    by 0x272DED: ??? (in /usr/bin/python3.10)
==167396==    by 0x26210D: ??? (in /usr/bin/python3.10)
==167396==    by 0x258A7A: _PyObject_MakeTpCall (in /usr/bin/python3.10)
==167396==    by 0x25214F: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==    by 0x2479C5: ??? (in /usr/bin/python3.10)
==167396==    by 0x33D255: PyEval_EvalCode (in /usr/bin/python3.10)
==167396==    by 0x342E2C: ??? (in /usr/bin/python3.10)
==167396==    by 0x262C58: ??? (in /usr/bin/python3.10)
==167396==    by 0x24D5D6: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396== 
==167396== Invalid write of size 8
==167396==    at 0x78981484: mkl_dft_avx2_scatter_z_z (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x621906A7: step1234 (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62191154: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x63B35282: __kmp_invoke_microtask (in /opt/intel/oneapi/compiler/2024.1/lib/libiomp5.so)
==167396==    by 0x63AC3972: __kmp_invoke_task_func (kmp_runtime.cpp:8487)
==167396==    by 0x63ABCDA9: __kmp_fork_call (kmp_runtime.cpp:2755)
==167396==    by 0x63A7E3C7: __kmpc_fork_call (kmp_csupport.cpp:379)
==167396==    by 0x62191060: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62190921: mkl_dft_z1d_via_2d_f (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x7895FD61: mkl_dft_avx2_compute_fwd_d_c2c_1d_o (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x78775EA3: compute_r2c_1d_even_fwd (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x5C8C63E9: DftiComputeForward (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_lp64.so.2)
==167396==  Address 0x57b4e9e0 is 1,856 bytes inside a block of size 2,141 free'd
==167396==    at 0x484DCD3: realloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==167396==    by 0x2694E1: ??? (in /usr/bin/python3.10)
==167396==    by 0x268D40: _PyUnicodeWriter_PrepareInternal (in /usr/bin/python3.10)
==167396==    by 0x275992: _PyUnicode_FormatAdvancedWriter (in /usr/bin/python3.10)
==167396==    by 0x27385D: ??? (in /usr/bin/python3.10)
==167396==    by 0x272CF2: ??? (in /usr/bin/python3.10)
==167396==    by 0x26210D: ??? (in /usr/bin/python3.10)
==167396==    by 0x258A7A: _PyObject_MakeTpCall (in /usr/bin/python3.10)
==167396==    by 0x25214F: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==    by 0x2479C5: ??? (in /usr/bin/python3.10)
==167396==    by 0x33D255: PyEval_EvalCode (in /usr/bin/python3.10)
==167396==    by 0x342E2C: ??? (in /usr/bin/python3.10)
==167396==  Block was alloc'd at
==167396==    at 0x484DCD3: realloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==167396==    by 0x2694E1: ??? (in /usr/bin/python3.10)
==167396==    by 0x268D40: _PyUnicodeWriter_PrepareInternal (in /usr/bin/python3.10)
==167396==    by 0x272DED: ??? (in /usr/bin/python3.10)
==167396==    by 0x26210D: ??? (in /usr/bin/python3.10)
==167396==    by 0x258A7A: _PyObject_MakeTpCall (in /usr/bin/python3.10)
==167396==    by 0x25214F: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==    by 0x2479C5: ??? (in /usr/bin/python3.10)
==167396==    by 0x33D255: PyEval_EvalCode (in /usr/bin/python3.10)
==167396==    by 0x342E2C: ??? (in /usr/bin/python3.10)
==167396==    by 0x262C58: ??? (in /usr/bin/python3.10)
==167396==    by 0x24D5D6: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396== 
==167396== Invalid write of size 8
==167396==    at 0x7898148A: mkl_dft_avx2_scatter_z_z (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x621906A7: step1234 (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62191154: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x63B35282: __kmp_invoke_microtask (in /opt/intel/oneapi/compiler/2024.1/lib/libiomp5.so)
==167396==    by 0x63AC3972: __kmp_invoke_task_func (kmp_runtime.cpp:8487)
==167396==    by 0x63ABCDA9: __kmp_fork_call (kmp_runtime.cpp:2755)
==167396==    by 0x63A7E3C7: __kmpc_fork_call (kmp_csupport.cpp:379)
==167396==    by 0x62191060: ttl_parallel_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x62190921: mkl_dft_z1d_via_2d_f (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x7895FD61: mkl_dft_avx2_compute_fwd_d_c2c_1d_o (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x78775EA3: compute_r2c_1d_even_fwd (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x5C8C63E9: DftiComputeForward (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_lp64.so.2)
==167396==  Address 0x57b4ea00 is 1,888 bytes inside a block of size 2,141 free'd
==167396==    at 0x484DCD3: realloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==167396==    by 0x2694E1: ??? (in /usr/bin/python3.10)
==167396==    by 0x268D40: _PyUnicodeWriter_PrepareInternal (in /usr/bin/python3.10)
==167396==    by 0x275992: _PyUnicode_FormatAdvancedWriter (in /usr/bin/python3.10)
==167396==    by 0x27385D: ??? (in /usr/bin/python3.10)
==167396==    by 0x272CF2: ??? (in /usr/bin/python3.10)
==167396==    by 0x26210D: ??? (in /usr/bin/python3.10)
==167396==    by 0x258A7A: _PyObject_MakeTpCall (in /usr/bin/python3.10)
==167396==    by 0x25214F: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==    by 0x2479C5: ??? (in /usr/bin/python3.10)
==167396==    by 0x33D255: PyEval_EvalCode (in /usr/bin/python3.10)
==167396==    by 0x342E2C: ??? (in /usr/bin/python3.10)
==167396==  Block was alloc'd at
==167396==    at 0x484DCD3: realloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==167396==    by 0x2694E1: ??? (in /usr/bin/python3.10)
==167396==    by 0x268D40: _PyUnicodeWriter_PrepareInternal (in /usr/bin/python3.10)
==167396==    by 0x272DED: ??? (in /usr/bin/python3.10)
==167396==    by 0x26210D: ??? (in /usr/bin/python3.10)
==167396==    by 0x258A7A: _PyObject_MakeTpCall (in /usr/bin/python3.10)
==167396==    by 0x25214F: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==    by 0x2479C5: ??? (in /usr/bin/python3.10)
==167396==    by 0x33D255: PyEval_EvalCode (in /usr/bin/python3.10)
==167396==    by 0x342E2C: ??? (in /usr/bin/python3.10)
==167396==    by 0x262C58: ??? (in /usr/bin/python3.10)
==167396==    by 0x24D5D6: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396== 
==167396== Invalid read of size 16
==167396==    at 0x78775C65: compute_task_fwd (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x6219134B: ttl_parallel_dynamic_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x63B35282: __kmp_invoke_microtask (in /opt/intel/oneapi/compiler/2024.1/lib/libiomp5.so)
==167396==    by 0x63AC3972: __kmp_invoke_task_func (kmp_runtime.cpp:8487)
==167396==    by 0x63ABCDA9: __kmp_fork_call (kmp_runtime.cpp:2755)
==167396==    by 0x63A7E3C7: __kmpc_fork_call (kmp_csupport.cpp:379)
==167396==    by 0x62191256: ttl_parallel_dynamic_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x78775F20: compute_r2c_1d_even_fwd (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x5C8C63E9: DftiComputeForward (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_lp64.so.2)
==167396==    by 0x5BCBD999: _forward_fft_R2C(std::vector<float, std::allocator<float> >) (mkl_fft.cpp:24)
==167396==    by 0x5BCBBCD3: __pyx_pf_10custom_fft_4MKL_forward_fft_R2C (fft.cpp:3671)
==167396==    by 0x5BCBBCD3: __pyx_pw_10custom_fft_5MKL_forward_fft_R2C(_object*, _object* const*, long, _object*) (fft.cpp:3629)
==167396==    by 0x24B26C: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==  Address 0x57b511e0 is 16 bytes before a block of size 1,941 alloc'd
==167396==    at 0x4848899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==167396==    by 0x2424D6: PyBytes_FromStringAndSize (in /usr/bin/python3.10)
==167396==    by 0x240C27: ??? (in /usr/bin/python3.10)
==167396==    by 0x240FB6: ??? (in /usr/bin/python3.10)
==167396==    by 0x24149C: ??? (in /usr/bin/python3.10)
==167396==    by 0x240FCB: ??? (in /usr/bin/python3.10)
==167396==    by 0x24151E: ??? (in /usr/bin/python3.10)
==167396==    by 0x240FCB: ??? (in /usr/bin/python3.10)
==167396==    by 0x33C87F: ??? (in /usr/bin/python3.10)
==167396==    by 0x34C894: ??? (in /usr/bin/python3.10)
==167396==    by 0x261B33: ??? (in /usr/bin/python3.10)
==167396==    by 0x250CF9: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396== 

valgrind: m_mallocfree.c:303 (get_bszB_as_is): Assertion 'bszB_lo == bszB_hi' failed.
valgrind: Heap block lo/hi size mismatch: lo = 32848, hi = 18446744073709551615.
This is probably caused by your program erroneously writing past the
end of a heap block and corrupting heap metadata.  If you fix any
invalid writes reported by Memcheck, this assertion failure will
probably go away.  Please try that before reporting this as a bug.


host stacktrace:
==167396==    at 0x5804284A: ??? (in /usr/libexec/valgrind/memcheck-amd64-linux)
==167396==    by 0x58042977: ??? (in /usr/libexec/valgrind/memcheck-amd64-linux)
==167396==    by 0x58042B1B: ??? (in /usr/libexec/valgrind/memcheck-amd64-linux)
==167396==    by 0x5804C8CF: ??? (in /usr/libexec/valgrind/memcheck-amd64-linux)
==167396==    by 0x5803AE9A: ??? (in /usr/libexec/valgrind/memcheck-amd64-linux)
==167396==    by 0x580395B7: ??? (in /usr/libexec/valgrind/memcheck-amd64-linux)
==167396==    by 0x5803DF3D: ??? (in /usr/libexec/valgrind/memcheck-amd64-linux)
==167396==    by 0x58038868: ??? (in /usr/libexec/valgrind/memcheck-amd64-linux)
==167396==    by 0x1015A53285: ???

sched status:
  running_tid=1

Thread 1: status = VgTs_Runnable (lwpid 167396)
==167396==    at 0x78775C7A: compute_task_fwd (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x6219134B: ttl_parallel_dynamic_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x63B35282: __kmp_invoke_microtask (in /opt/intel/oneapi/compiler/2024.1/lib/libiomp5.so)
==167396==    by 0x63AC3972: __kmp_invoke_task_func (kmp_runtime.cpp:8487)
==167396==    by 0x63ABCDA9: __kmp_fork_call (kmp_runtime.cpp:2755)
==167396==    by 0x63A7E3C7: __kmpc_fork_call (kmp_csupport.cpp:379)
==167396==    by 0x62191256: ttl_parallel_dynamic_team (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_thread.so.2)
==167396==    by 0x78775F20: compute_r2c_1d_even_fwd (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_avx2.so.2)
==167396==    by 0x5C8C63E9: DftiComputeForward (in /opt/intel/oneapi/mkl/2024.1/lib/libmkl_intel_lp64.so.2)
==167396==    by 0x5BCBD999: _forward_fft_R2C(std::vector<float, std::allocator<float> >) (mkl_fft.cpp:24)
==167396==    by 0x5BCBBCD3: __pyx_pf_10custom_fft_4MKL_forward_fft_R2C (fft.cpp:3671)
==167396==    by 0x5BCBBCD3: __pyx_pw_10custom_fft_5MKL_forward_fft_R2C(_object*, _object* const*, long, _object*) (fft.cpp:3629)
==167396==    by 0x24B26C: _PyEval_EvalFrameDefault (in /usr/bin/python3.10)
==167396==    by 0x2479C5: ??? (in /usr/bin/python3.10)
==167396==    by 0x33D255: PyEval_EvalCode (in /usr/bin/python3.10)
==167396==    by 0x368107: ??? (in /usr/bin/python3.10)
==167396==    by 0x3619CA: ??? (in /usr/bin/python3.10)
==167396==    by 0x367E54: ??? (in /usr/bin/python3.10)
==167396==    by 0x367337: _PyRun_SimpleFileObject (in /usr/bin/python3.10)
==167396==    by 0x366F82: _PyRun_AnyFileObject (in /usr/bin/python3.10)
==167396==    by 0x359A5D: Py_RunMain (in /usr/bin/python3.10)
==167396==    by 0x33002C: Py_BytesMain (in /usr/bin/python3.10)
==167396==    by 0x49C4D8F: (below main) (libc_start_call_main.h:58)
client stack range: [0x1FFEFEC000 0x1FFF000FFF] client SP: 0x1FFEFFD2F0
valgrind stack range: [0x1002CBA000 0x1002DB9FFF] top usage: 18984 of 1048576

Thread 2: status = VgTs_WaitSys syscall 202 (lwpid 167443)
==167396==    at 0x4A2C117: __futex_abstimed_wait_common64 (futex-internal.c:57)
==167396==    by 0x4A2C117: __futex_abstimed_wait_common (futex-internal.c:87)
==167396==    by 0x4A2C117: __futex_abstimed_wait_cancelable64 (futex-internal.c:139)
==167396==    by 0x4A2EA40: __pthread_cond_wait_common (pthread_cond_wait.c:503)
==167396==    by 0x4A2EA40: pthread_cond_wait@@GLIBC_2.3.2 (pthread_cond_wait.c:627)
==167396==    by 0x679E6FA: blas_thread_server (in /home/bl4z3/.local/lib/python3.10/site-packages/numpy.libs/libopenblas64_p-r0-0cf96a72.3.23.dev.so)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x8E3E000 0x963CFFF] client SP: 0x963CCF0
valgrind stack range: [0x10065CB000 0x10066CAFFF] top usage: 5472 of 1048576

Thread 3: status = VgTs_WaitSys syscall 202 (lwpid 167444)
==167396==    at 0x4A2C117: __futex_abstimed_wait_common64 (futex-internal.c:57)
==167396==    by 0x4A2C117: __futex_abstimed_wait_common (futex-internal.c:87)
==167396==    by 0x4A2C117: __futex_abstimed_wait_cancelable64 (futex-internal.c:139)
==167396==    by 0x4A2EA40: __pthread_cond_wait_common (pthread_cond_wait.c:503)
==167396==    by 0x4A2EA40: pthread_cond_wait@@GLIBC_2.3.2 (pthread_cond_wait.c:627)
==167396==    by 0x679E6FA: blas_thread_server (in /home/bl4z3/.local/lib/python3.10/site-packages/numpy.libs/libopenblas64_p-r0-0cf96a72.3.23.dev.so)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x963F000 0x9E3DFFF] client SP: 0x9E3DCF0
valgrind stack range: [0x1006883000 0x1006982FFF] top usage: 5472 of 1048576

Thread 4: status = VgTs_WaitSys syscall 202 (lwpid 167445)
==167396==    at 0x4A2C117: __futex_abstimed_wait_common64 (futex-internal.c:57)
==167396==    by 0x4A2C117: __futex_abstimed_wait_common (futex-internal.c:87)
==167396==    by 0x4A2C117: __futex_abstimed_wait_cancelable64 (futex-internal.c:139)
==167396==    by 0x4A2EA40: __pthread_cond_wait_common (pthread_cond_wait.c:503)
==167396==    by 0x4A2EA40: pthread_cond_wait@@GLIBC_2.3.2 (pthread_cond_wait.c:627)
==167396==    by 0x679E6FA: blas_thread_server (in /home/bl4z3/.local/lib/python3.10/site-packages/numpy.libs/libopenblas64_p-r0-0cf96a72.3.23.dev.so)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0xDE40000 0xE63EFFF] client SP: 0xE63ECF0
valgrind stack range: [0x1006B8F000 0x1006C8EFFF] top usage: 5472 of 1048576

Thread 5: status = VgTs_WaitSys syscall 202 (lwpid 167451)
==167396==    at 0x4A2C117: __futex_abstimed_wait_common64 (futex-internal.c:57)
==167396==    by 0x4A2C117: __futex_abstimed_wait_common (futex-internal.c:87)
==167396==    by 0x4A2C117: __futex_abstimed_wait_cancelable64 (futex-internal.c:139)
==167396==    by 0x4A2EA40: __pthread_cond_wait_common (pthread_cond_wait.c:503)
==167396==    by 0x4A2EA40: pthread_cond_wait@@GLIBC_2.3.2 (pthread_cond_wait.c:627)
==167396==    by 0x679E6FA: blas_thread_server (in /home/bl4z3/.local/lib/python3.10/site-packages/numpy.libs/libopenblas64_p-r0-0cf96a72.3.23.dev.so)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x10641000 0x10E3FFFF] client SP: 0x10E3FCF0
valgrind stack range: [0x1006E97000 0x1006F96FFF] top usage: 5472 of 1048576

Thread 6: status = VgTs_WaitSys syscall 202 (lwpid 167452)
==167396==    at 0x4A2C117: __futex_abstimed_wait_common64 (futex-internal.c:57)
==167396==    by 0x4A2C117: __futex_abstimed_wait_common (futex-internal.c:87)
==167396==    by 0x4A2C117: __futex_abstimed_wait_cancelable64 (futex-internal.c:139)
==167396==    by 0x4A2EA40: __pthread_cond_wait_common (pthread_cond_wait.c:503)
==167396==    by 0x4A2EA40: pthread_cond_wait@@GLIBC_2.3.2 (pthread_cond_wait.c:627)
==167396==    by 0x679E6FA: blas_thread_server (in /home/bl4z3/.local/lib/python3.10/site-packages/numpy.libs/libopenblas64_p-r0-0cf96a72.3.23.dev.so)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x12E42000 0x13640FFF] client SP: 0x13640CF0
valgrind stack range: [0x100719F000 0x100729EFFF] top usage: 5472 of 1048576

Thread 7: status = VgTs_WaitSys syscall 202 (lwpid 167453)
==167396==    at 0x4A2C117: __futex_abstimed_wait_common64 (futex-internal.c:57)
==167396==    by 0x4A2C117: __futex_abstimed_wait_common (futex-internal.c:87)
==167396==    by 0x4A2C117: __futex_abstimed_wait_cancelable64 (futex-internal.c:139)
==167396==    by 0x4A2EA40: __pthread_cond_wait_common (pthread_cond_wait.c:503)
==167396==    by 0x4A2EA40: pthread_cond_wait@@GLIBC_2.3.2 (pthread_cond_wait.c:627)
==167396==    by 0x679E6FA: blas_thread_server (in /home/bl4z3/.local/lib/python3.10/site-packages/numpy.libs/libopenblas64_p-r0-0cf96a72.3.23.dev.so)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x13643000 0x13E41FFF] client SP: 0x13E41CF0
valgrind stack range: [0x10074A3000 0x10075A2FFF] top usage: 5472 of 1048576

Thread 8: status = VgTs_WaitSys syscall 202 (lwpid 167454)
==167396==    at 0x4A2C117: __futex_abstimed_wait_common64 (futex-internal.c:57)
==167396==    by 0x4A2C117: __futex_abstimed_wait_common (futex-internal.c:87)
==167396==    by 0x4A2C117: __futex_abstimed_wait_cancelable64 (futex-internal.c:139)
==167396==    by 0x4A2EA40: __pthread_cond_wait_common (pthread_cond_wait.c:503)
==167396==    by 0x4A2EA40: pthread_cond_wait@@GLIBC_2.3.2 (pthread_cond_wait.c:627)
==167396==    by 0x679E6FA: blas_thread_server (in /home/bl4z3/.local/lib/python3.10/site-packages/numpy.libs/libopenblas64_p-r0-0cf96a72.3.23.dev.so)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x17E44000 0x18642FFF] client SP: 0x18642CF0
valgrind stack range: [0x10077AF000 0x10078AEFFF] top usage: 5472 of 1048576

Thread 9: status = VgTs_WaitSys syscall 202 (lwpid 167455)
==167396==    at 0x4A2C117: __futex_abstimed_wait_common64 (futex-internal.c:57)
==167396==    by 0x4A2C117: __futex_abstimed_wait_common (futex-internal.c:87)
==167396==    by 0x4A2C117: __futex_abstimed_wait_cancelable64 (futex-internal.c:139)
==167396==    by 0x4A2EA40: __pthread_cond_wait_common (pthread_cond_wait.c:503)
==167396==    by 0x4A2EA40: pthread_cond_wait@@GLIBC_2.3.2 (pthread_cond_wait.c:627)
==167396==    by 0x679E6FA: blas_thread_server (in /home/bl4z3/.local/lib/python3.10/site-packages/numpy.libs/libopenblas64_p-r0-0cf96a72.3.23.dev.so)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x18645000 0x18E43FFF] client SP: 0x18E43CF0
valgrind stack range: [0x1007AB3000 0x1007BB2FFF] top usage: 5472 of 1048576

Thread 10: status = VgTs_WaitSys syscall 202 (lwpid 167456)
==167396==    at 0x4A2C117: __futex_abstimed_wait_common64 (futex-internal.c:57)
==167396==    by 0x4A2C117: __futex_abstimed_wait_common (futex-internal.c:87)
==167396==    by 0x4A2C117: __futex_abstimed_wait_cancelable64 (futex-internal.c:139)
==167396==    by 0x4A2EA40: __pthread_cond_wait_common (pthread_cond_wait.c:503)
==167396==    by 0x4A2EA40: pthread_cond_wait@@GLIBC_2.3.2 (pthread_cond_wait.c:627)
==167396==    by 0x679E6FA: blas_thread_server (in /home/bl4z3/.local/lib/python3.10/site-packages/numpy.libs/libopenblas64_p-r0-0cf96a72.3.23.dev.so)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x18E46000 0x19644FFF] client SP: 0x19644CF0
valgrind stack range: [0x1007DB7000 0x1007EB6FFF] top usage: 5472 of 1048576

Thread 11: status = VgTs_WaitSys syscall 202 (lwpid 167457)
==167396==    at 0x4A2C117: __futex_abstimed_wait_common64 (futex-internal.c:57)
==167396==    by 0x4A2C117: __futex_abstimed_wait_common (futex-internal.c:87)
==167396==    by 0x4A2C117: __futex_abstimed_wait_cancelable64 (futex-internal.c:139)
==167396==    by 0x4A2EA40: __pthread_cond_wait_common (pthread_cond_wait.c:503)
==167396==    by 0x4A2EA40: pthread_cond_wait@@GLIBC_2.3.2 (pthread_cond_wait.c:627)
==167396==    by 0x679E6FA: blas_thread_server (in /home/bl4z3/.local/lib/python3.10/site-packages/numpy.libs/libopenblas64_p-r0-0cf96a72.3.23.dev.so)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x1D647000 0x1DE45FFF] client SP: 0x1DE45CF0
valgrind stack range: [0x10080C3000 0x10081C2FFF] top usage: 5472 of 1048576

Thread 12: status = VgTs_WaitSys syscall 202 (lwpid 167458)
==167396==    at 0x4A2C117: __futex_abstimed_wait_common64 (futex-internal.c:57)
==167396==    by 0x4A2C117: __futex_abstimed_wait_common (futex-internal.c:87)
==167396==    by 0x4A2C117: __futex_abstimed_wait_cancelable64 (futex-internal.c:139)
==167396==    by 0x4A2EA40: __pthread_cond_wait_common (pthread_cond_wait.c:503)
==167396==    by 0x4A2EA40: pthread_cond_wait@@GLIBC_2.3.2 (pthread_cond_wait.c:627)
==167396==    by 0x679E6FA: blas_thread_server (in /home/bl4z3/.local/lib/python3.10/site-packages/numpy.libs/libopenblas64_p-r0-0cf96a72.3.23.dev.so)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x1DE48000 0x1E646FFF] client SP: 0x1E646CF0
valgrind stack range: [0x10083C7000 0x10084C6FFF] top usage: 5472 of 1048576

Thread 13: status = VgTs_WaitSys syscall 202 (lwpid 167459)
==167396==    at 0x4A2C117: __futex_abstimed_wait_common64 (futex-internal.c:57)
==167396==    by 0x4A2C117: __futex_abstimed_wait_common (futex-internal.c:87)
==167396==    by 0x4A2C117: __futex_abstimed_wait_cancelable64 (futex-internal.c:139)
==167396==    by 0x4A2EA40: __pthread_cond_wait_common (pthread_cond_wait.c:503)
==167396==    by 0x4A2EA40: pthread_cond_wait@@GLIBC_2.3.2 (pthread_cond_wait.c:627)
==167396==    by 0x679E6FA: blas_thread_server (in /home/bl4z3/.local/lib/python3.10/site-packages/numpy.libs/libopenblas64_p-r0-0cf96a72.3.23.dev.so)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x24649000 0x24E47FFF] client SP: 0x24E47CF0
valgrind stack range: [0x10086D7000 0x10087D6FFF] top usage: 5472 of 1048576

Thread 14: status = VgTs_WaitSys syscall 202 (lwpid 167460)
==167396==    at 0x4A2C117: __futex_abstimed_wait_common64 (futex-internal.c:57)
==167396==    by 0x4A2C117: __futex_abstimed_wait_common (futex-internal.c:87)
==167396==    by 0x4A2C117: __futex_abstimed_wait_cancelable64 (futex-internal.c:139)
==167396==    by 0x4A2EA40: __pthread_cond_wait_common (pthread_cond_wait.c:503)
==167396==    by 0x4A2EA40: pthread_cond_wait@@GLIBC_2.3.2 (pthread_cond_wait.c:627)
==167396==    by 0x679E6FA: blas_thread_server (in /home/bl4z3/.local/lib/python3.10/site-packages/numpy.libs/libopenblas64_p-r0-0cf96a72.3.23.dev.so)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x26E4A000 0x27648FFF] client SP: 0x27648CF0
valgrind stack range: [0x10089DF000 0x1008ADEFFF] top usage: 5472 of 1048576

Thread 15: status = VgTs_WaitSys syscall 202 (lwpid 167461)
==167396==    at 0x4A2C117: __futex_abstimed_wait_common64 (futex-internal.c:57)
==167396==    by 0x4A2C117: __futex_abstimed_wait_common (futex-internal.c:87)
==167396==    by 0x4A2C117: __futex_abstimed_wait_cancelable64 (futex-internal.c:139)
==167396==    by 0x4A2EA40: __pthread_cond_wait_common (pthread_cond_wait.c:503)
==167396==    by 0x4A2EA40: pthread_cond_wait@@GLIBC_2.3.2 (pthread_cond_wait.c:627)
==167396==    by 0x679E6FA: blas_thread_server (in /home/bl4z3/.local/lib/python3.10/site-packages/numpy.libs/libopenblas64_p-r0-0cf96a72.3.23.dev.so)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x2964B000 0x29E49FFF] client SP: 0x29E49CF0
valgrind stack range: [0x1008CE7000 0x1008DE6FFF] top usage: 5472 of 1048576

Thread 16: status = VgTs_WaitSys syscall 202 (lwpid 167462)
==167396==    at 0x4A2C117: __futex_abstimed_wait_common64 (futex-internal.c:57)
==167396==    by 0x4A2C117: __futex_abstimed_wait_common (futex-internal.c:87)
==167396==    by 0x4A2C117: __futex_abstimed_wait_cancelable64 (futex-internal.c:139)
==167396==    by 0x4A2EA40: __pthread_cond_wait_common (pthread_cond_wait.c:503)
==167396==    by 0x4A2EA40: pthread_cond_wait@@GLIBC_2.3.2 (pthread_cond_wait.c:627)
==167396==    by 0x679E6FA: blas_thread_server (in /home/bl4z3/.local/lib/python3.10/site-packages/numpy.libs/libopenblas64_p-r0-0cf96a72.3.23.dev.so)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x2BE4C000 0x2C64AFFF] client SP: 0x2C64ACF0
valgrind stack range: [0x1008FEF000 0x10090EEFFF] top usage: 5472 of 1048576

Thread 17: status = VgTs_WaitSys syscall 202 (lwpid 167463)
==167396==    at 0x4A2C117: __futex_abstimed_wait_common64 (futex-internal.c:57)
==167396==    by 0x4A2C117: __futex_abstimed_wait_common (futex-internal.c:87)
==167396==    by 0x4A2C117: __futex_abstimed_wait_cancelable64 (futex-internal.c:139)
==167396==    by 0x4A2EA40: __pthread_cond_wait_common (pthread_cond_wait.c:503)
==167396==    by 0x4A2EA40: pthread_cond_wait@@GLIBC_2.3.2 (pthread_cond_wait.c:627)
==167396==    by 0x679E6FA: blas_thread_server (in /home/bl4z3/.local/lib/python3.10/site-packages/numpy.libs/libopenblas64_p-r0-0cf96a72.3.23.dev.so)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x2E64D000 0x2EE4BFFF] client SP: 0x2EE4BCF0
valgrind stack range: [0x10092F7000 0x10093F6FFF] top usage: 5472 of 1048576

Thread 18: status = VgTs_WaitSys syscall 202 (lwpid 167464)
==167396==    at 0x4A2C117: __futex_abstimed_wait_common64 (futex-internal.c:57)
==167396==    by 0x4A2C117: __futex_abstimed_wait_common (futex-internal.c:87)
==167396==    by 0x4A2C117: __futex_abstimed_wait_cancelable64 (futex-internal.c:139)
==167396==    by 0x4A2EA40: __pthread_cond_wait_common (pthread_cond_wait.c:503)
==167396==    by 0x4A2EA40: pthread_cond_wait@@GLIBC_2.3.2 (pthread_cond_wait.c:627)
==167396==    by 0x679E6FA: blas_thread_server (in /home/bl4z3/.local/lib/python3.10/site-packages/numpy.libs/libopenblas64_p-r0-0cf96a72.3.23.dev.so)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x30E4E000 0x3164CFFF] client SP: 0x3164CCF0
valgrind stack range: [0x10095FF000 0x10096FEFFF] top usage: 5472 of 1048576

Thread 19: status = VgTs_WaitSys syscall 202 (lwpid 167465)
==167396==    at 0x4A2C117: __futex_abstimed_wait_common64 (futex-internal.c:57)
==167396==    by 0x4A2C117: __futex_abstimed_wait_common (futex-internal.c:87)
==167396==    by 0x4A2C117: __futex_abstimed_wait_cancelable64 (futex-internal.c:139)
==167396==    by 0x4A2EA40: __pthread_cond_wait_common (pthread_cond_wait.c:503)
==167396==    by 0x4A2EA40: pthread_cond_wait@@GLIBC_2.3.2 (pthread_cond_wait.c:627)
==167396==    by 0x679E6FA: blas_thread_server (in /home/bl4z3/.local/lib/python3.10/site-packages/numpy.libs/libopenblas64_p-r0-0cf96a72.3.23.dev.so)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x3364F000 0x33E4DFFF] client SP: 0x33E4DCF0
valgrind stack range: [0x1009907000 0x1009A06FFF] top usage: 5472 of 1048576

Thread 20: status = VgTs_WaitSys syscall 202 (lwpid 167467)
==167396==    at 0x4A2C117: __futex_abstimed_wait_common64 (futex-internal.c:57)
==167396==    by 0x4A2C117: __futex_abstimed_wait_common (futex-internal.c:87)
==167396==    by 0x4A2C117: __futex_abstimed_wait_cancelable64 (futex-internal.c:139)
==167396==    by 0x4A2EA40: __pthread_cond_wait_common (pthread_cond_wait.c:503)
==167396==    by 0x4A2EA40: pthread_cond_wait@@GLIBC_2.3.2 (pthread_cond_wait.c:627)
==167396==    by 0x679E6FA: blas_thread_server (in /home/bl4z3/.local/lib/python3.10/site-packages/numpy.libs/libopenblas64_p-r0-0cf96a72.3.23.dev.so)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x35E50000 0x3664EFFF] client SP: 0x3664ECF0
valgrind stack range: [0x1009C0F000 0x1009D0EFFF] top usage: 5472 of 1048576

Thread 21: status = VgTs_WaitSys syscall 202 (lwpid 167468)
==167396==    at 0x4A2C117: __futex_abstimed_wait_common64 (futex-internal.c:57)
==167396==    by 0x4A2C117: __futex_abstimed_wait_common (futex-internal.c:87)
==167396==    by 0x4A2C117: __futex_abstimed_wait_cancelable64 (futex-internal.c:139)
==167396==    by 0x4A2EA40: __pthread_cond_wait_common (pthread_cond_wait.c:503)
==167396==    by 0x4A2EA40: pthread_cond_wait@@GLIBC_2.3.2 (pthread_cond_wait.c:627)
==167396==    by 0x679E6FA: blas_thread_server (in /home/bl4z3/.local/lib/python3.10/site-packages/numpy.libs/libopenblas64_p-r0-0cf96a72.3.23.dev.so)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x38651000 0x38E4FFFF] client SP: 0x38E4FCF0
valgrind stack range: [0x1009F17000 0x100A016FFF] top usage: 5472 of 1048576

Thread 22: status = VgTs_WaitSys syscall 202 (lwpid 167469)
==167396==    at 0x4A2C117: __futex_abstimed_wait_common64 (futex-internal.c:57)
==167396==    by 0x4A2C117: __futex_abstimed_wait_common (futex-internal.c:87)
==167396==    by 0x4A2C117: __futex_abstimed_wait_cancelable64 (futex-internal.c:139)
==167396==    by 0x4A2EA40: __pthread_cond_wait_common (pthread_cond_wait.c:503)
==167396==    by 0x4A2EA40: pthread_cond_wait@@GLIBC_2.3.2 (pthread_cond_wait.c:627)
==167396==    by 0x679E6FA: blas_thread_server (in /home/bl4z3/.local/lib/python3.10/site-packages/numpy.libs/libopenblas64_p-r0-0cf96a72.3.23.dev.so)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x38E52000 0x39650FFF] client SP: 0x39650CF0
valgrind stack range: [0x100A21B000 0x100A31AFFF] top usage: 5472 of 1048576

Thread 23: status = VgTs_WaitSys syscall 202 (lwpid 167470)
==167396==    at 0x4A2C117: __futex_abstimed_wait_common64 (futex-internal.c:57)
==167396==    by 0x4A2C117: __futex_abstimed_wait_common (futex-internal.c:87)
==167396==    by 0x4A2C117: __futex_abstimed_wait_cancelable64 (futex-internal.c:139)
==167396==    by 0x4A2EA40: __pthread_cond_wait_common (pthread_cond_wait.c:503)
==167396==    by 0x4A2EA40: pthread_cond_wait@@GLIBC_2.3.2 (pthread_cond_wait.c:627)
==167396==    by 0x679E6FA: blas_thread_server (in /home/bl4z3/.local/lib/python3.10/site-packages/numpy.libs/libopenblas64_p-r0-0cf96a72.3.23.dev.so)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x39653000 0x39E51FFF] client SP: 0x39E51CF0
valgrind stack range: [0x100A51F000 0x100A61EFFF] top usage: 5472 of 1048576

Thread 24: status = VgTs_WaitSys syscall 202 (lwpid 167471)
==167396==    at 0x4A2C117: __futex_abstimed_wait_common64 (futex-internal.c:57)
==167396==    by 0x4A2C117: __futex_abstimed_wait_common (futex-internal.c:87)
==167396==    by 0x4A2C117: __futex_abstimed_wait_cancelable64 (futex-internal.c:139)
==167396==    by 0x4A2EA40: __pthread_cond_wait_common (pthread_cond_wait.c:503)
==167396==    by 0x4A2EA40: pthread_cond_wait@@GLIBC_2.3.2 (pthread_cond_wait.c:627)
==167396==    by 0x679E6FA: blas_thread_server (in /home/bl4z3/.local/lib/python3.10/site-packages/numpy.libs/libopenblas64_p-r0-0cf96a72.3.23.dev.so)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x3FE54000 0x40652FFF] client SP: 0x40652CF0
valgrind stack range: [0x100A82F000 0x100A92EFFF] top usage: 5472 of 1048576

Thread 25: status = VgTs_WaitSys syscall 202 (lwpid 167472)
==167396==    at 0x4A2C117: __futex_abstimed_wait_common64 (futex-internal.c:57)
==167396==    by 0x4A2C117: __futex_abstimed_wait_common (futex-internal.c:87)
==167396==    by 0x4A2C117: __futex_abstimed_wait_cancelable64 (futex-internal.c:139)
==167396==    by 0x4A2EA40: __pthread_cond_wait_common (pthread_cond_wait.c:503)
==167396==    by 0x4A2EA40: pthread_cond_wait@@GLIBC_2.3.2 (pthread_cond_wait.c:627)
==167396==    by 0x679E6FA: blas_thread_server (in /home/bl4z3/.local/lib/python3.10/site-packages/numpy.libs/libopenblas64_p-r0-0cf96a72.3.23.dev.so)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x42655000 0x42E53FFF] client SP: 0x42E53CF0
valgrind stack range: [0x100AB37000 0x100AC36FFF] top usage: 5472 of 1048576

Thread 26: status = VgTs_WaitSys syscall 202 (lwpid 167473)
==167396==    at 0x4A2C117: __futex_abstimed_wait_common64 (futex-internal.c:57)
==167396==    by 0x4A2C117: __futex_abstimed_wait_common (futex-internal.c:87)
==167396==    by 0x4A2C117: __futex_abstimed_wait_cancelable64 (futex-internal.c:139)
==167396==    by 0x4A2EA40: __pthread_cond_wait_common (pthread_cond_wait.c:503)
==167396==    by 0x4A2EA40: pthread_cond_wait@@GLIBC_2.3.2 (pthread_cond_wait.c:627)
==167396==    by 0x679E6FA: blas_thread_server (in /home/bl4z3/.local/lib/python3.10/site-packages/numpy.libs/libopenblas64_p-r0-0cf96a72.3.23.dev.so)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x44E56000 0x45654FFF] client SP: 0x45654CF0
valgrind stack range: [0x100AE3F000 0x100AF3EFFF] top usage: 5472 of 1048576

Thread 27: status = VgTs_WaitSys syscall 202 (lwpid 167474)
==167396==    at 0x4A2C117: __futex_abstimed_wait_common64 (futex-internal.c:57)
==167396==    by 0x4A2C117: __futex_abstimed_wait_common (futex-internal.c:87)
==167396==    by 0x4A2C117: __futex_abstimed_wait_cancelable64 (futex-internal.c:139)
==167396==    by 0x4A2EA40: __pthread_cond_wait_common (pthread_cond_wait.c:503)
==167396==    by 0x4A2EA40: pthread_cond_wait@@GLIBC_2.3.2 (pthread_cond_wait.c:627)
==167396==    by 0x679E6FA: blas_thread_server (in /home/bl4z3/.local/lib/python3.10/site-packages/numpy.libs/libopenblas64_p-r0-0cf96a72.3.23.dev.so)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x47657000 0x47E55FFF] client SP: 0x47E55CF0
valgrind stack range: [0x100B147000 0x100B246FFF] top usage: 5472 of 1048576

Thread 28: status = VgTs_WaitSys syscall 202 (lwpid 167475)
==167396==    at 0x4A2C117: __futex_abstimed_wait_common64 (futex-internal.c:57)
==167396==    by 0x4A2C117: __futex_abstimed_wait_common (futex-internal.c:87)
==167396==    by 0x4A2C117: __futex_abstimed_wait_cancelable64 (futex-internal.c:139)
==167396==    by 0x4A2EA40: __pthread_cond_wait_common (pthread_cond_wait.c:503)
==167396==    by 0x4A2EA40: pthread_cond_wait@@GLIBC_2.3.2 (pthread_cond_wait.c:627)
==167396==    by 0x679E6FA: blas_thread_server (in /home/bl4z3/.local/lib/python3.10/site-packages/numpy.libs/libopenblas64_p-r0-0cf96a72.3.23.dev.so)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x47E58000 0x48656FFF] client SP: 0x48656CF0
valgrind stack range: [0x100B44B000 0x100B54AFFF] top usage: 5472 of 1048576

Thread 29: status = VgTs_WaitSys syscall 202 (lwpid 167476)
==167396==    at 0x4A2C117: __futex_abstimed_wait_common64 (futex-internal.c:57)
==167396==    by 0x4A2C117: __futex_abstimed_wait_common (futex-internal.c:87)
==167396==    by 0x4A2C117: __futex_abstimed_wait_cancelable64 (futex-internal.c:139)
==167396==    by 0x4A2EA40: __pthread_cond_wait_common (pthread_cond_wait.c:503)
==167396==    by 0x4A2EA40: pthread_cond_wait@@GLIBC_2.3.2 (pthread_cond_wait.c:627)
==167396==    by 0x679E6FA: blas_thread_server (in /home/bl4z3/.local/lib/python3.10/site-packages/numpy.libs/libopenblas64_p-r0-0cf96a72.3.23.dev.so)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x4C659000 0x4CE57FFF] client SP: 0x4CE57CF0
valgrind stack range: [0x100B757000 0x100B856FFF] top usage: 5472 of 1048576

Thread 30: status = VgTs_WaitSys syscall 202 (lwpid 167477)
==167396==    at 0x4A2C117: __futex_abstimed_wait_common64 (futex-internal.c:57)
==167396==    by 0x4A2C117: __futex_abstimed_wait_common (futex-internal.c:87)
==167396==    by 0x4A2C117: __futex_abstimed_wait_cancelable64 (futex-internal.c:139)
==167396==    by 0x4A2EA40: __pthread_cond_wait_common (pthread_cond_wait.c:503)
==167396==    by 0x4A2EA40: pthread_cond_wait@@GLIBC_2.3.2 (pthread_cond_wait.c:627)
==167396==    by 0x679E6FA: blas_thread_server (in /home/bl4z3/.local/lib/python3.10/site-packages/numpy.libs/libopenblas64_p-r0-0cf96a72.3.23.dev.so)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x4CE5A000 0x4D658FFF] client SP: 0x4D658CF0
valgrind stack range: [0x100BA5B000 0x100BB5AFFF] top usage: 5472 of 1048576

Thread 31: status = VgTs_WaitSys syscall 202 (lwpid 167478)
==167396==    at 0x4A2C117: __futex_abstimed_wait_common64 (futex-internal.c:57)
==167396==    by 0x4A2C117: __futex_abstimed_wait_common (futex-internal.c:87)
==167396==    by 0x4A2C117: __futex_abstimed_wait_cancelable64 (futex-internal.c:139)
==167396==    by 0x4A2EA40: __pthread_cond_wait_common (pthread_cond_wait.c:503)
==167396==    by 0x4A2EA40: pthread_cond_wait@@GLIBC_2.3.2 (pthread_cond_wait.c:627)
==167396==    by 0x679E6FA: blas_thread_server (in /home/bl4z3/.local/lib/python3.10/site-packages/numpy.libs/libopenblas64_p-r0-0cf96a72.3.23.dev.so)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x5165B000 0x51E59FFF] client SP: 0x51E59CF0
valgrind stack range: [0x100BD67000 0x100BE66FFF] top usage: 5472 of 1048576

Thread 32: status = VgTs_WaitSys syscall 202 (lwpid 167479)
==167396==    at 0x4A2C117: __futex_abstimed_wait_common64 (futex-internal.c:57)
==167396==    by 0x4A2C117: __futex_abstimed_wait_common (futex-internal.c:87)
==167396==    by 0x4A2C117: __futex_abstimed_wait_cancelable64 (futex-internal.c:139)
==167396==    by 0x4A2EA40: __pthread_cond_wait_common (pthread_cond_wait.c:503)
==167396==    by 0x4A2EA40: pthread_cond_wait@@GLIBC_2.3.2 (pthread_cond_wait.c:627)
==167396==    by 0x679E6FA: blas_thread_server (in /home/bl4z3/.local/lib/python3.10/site-packages/numpy.libs/libopenblas64_p-r0-0cf96a72.3.23.dev.so)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x53E5C000 0x5465AFFF] client SP: 0x5465ACF0
valgrind stack range: [0x100C06F000 0x100C16EFFF] top usage: 5472 of 1048576

Thread 33: status = VgTs_Yielding (lwpid 167614)
==167396==    at 0x63A7BE92: __kmp_wait_template<kmp_flag_64<false, true>, true, false, true> (kmp_wait_release.h:610)
==167396==    by 0x63A7BE92: kmp_flag_64<false, true>::wait(kmp_info*, int, void*) (kmp_wait_release.h:978)
==167396==    by 0x63A762BE: __kmp_hyper_barrier_release(barrier_type, kmp_info*, int, int, int, void*) (kmp_barrier.cpp:1191)
==167396==    by 0x63A79E56: __kmp_fork_barrier(int, int) (kmp_barrier.cpp:2547)
==167396==    by 0x63AC233C: __kmp_launch_thread (kmp_runtime.cpp:6712)
==167396==    by 0x63B36420: __kmp_launch_worker(void*) (z_Linux_util.cpp:560)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x7BB0B000 0x7C309FFF] client SP: 0x7C3099F0
valgrind stack range: [0x1016BC6000 0x1016CC5FFF] top usage: 4360 of 1048576

Thread 34: status = VgTs_Yielding (lwpid 167615)
==167396==    at 0x63A7BE92: __kmp_wait_template<kmp_flag_64<false, true>, true, false, true> (kmp_wait_release.h:610)
==167396==    by 0x63A7BE92: kmp_flag_64<false, true>::wait(kmp_info*, int, void*) (kmp_wait_release.h:978)
==167396==    by 0x63A762BE: __kmp_hyper_barrier_release(barrier_type, kmp_info*, int, int, int, void*) (kmp_barrier.cpp:1191)
==167396==    by 0x63A79E56: __kmp_fork_barrier(int, int) (kmp_barrier.cpp:2547)
==167396==    by 0x63AC233C: __kmp_launch_thread (kmp_runtime.cpp:6712)
==167396==    by 0x63B36420: __kmp_launch_worker(void*) (z_Linux_util.cpp:560)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
client stack range: [0x7C30D000 0x7CB0BFFF] client SP: 0x7CB0BA30
valgrind stack range: [0x1016ECA000 0x1016FC9FFF] top usage: 4360 of 1048576

Thread 35: status = VgTs_WaitSys syscall 24 (lwpid 167616)
==167396==    at 0x4AA3C9B: sched_yield (syscall-template.S:120)
==167396==    by 0x63A7BEDD: __kmp_wait_template<kmp_flag_64<false, true>, true, false, true> (kmp_wait_release.h:610)
==167396==    by 0x63A7BEDD: kmp_flag_64<false, true>::wait(kmp_info*, int, void*) (kmp_wait_release.h:978)
==167396==    by 0x63A762BE: __kmp_hyper_barrier_release(barrier_type, kmp_info*, int, int, int, void*) (kmp_barrier.cpp:1191)
==167396==    by 0x63A79E56: __kmp_fork_barrier(int, int) (kmp_barrier.cpp:2547)
==167396==    by 0x63AC233C: __kmp_launch_thread (kmp_runtime.cpp:6712)
==167396==    by 0x63B36420: __kmp_launch_worker(void*) (z_Linux_util.cpp:560)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x7CB0F000 0x7D30EFFF] client SP: 0x7D30DA68
valgrind stack range: [0x10171D2000 0x10172D1FFF] top usage: 4360 of 1048576

Thread 36: status = VgTs_Yielding (lwpid 167617)
==167396==    at 0x63A7BE92: __kmp_wait_template<kmp_flag_64<false, true>, true, false, true> (kmp_wait_release.h:610)
==167396==    by 0x63A7BE92: kmp_flag_64<false, true>::wait(kmp_info*, int, void*) (kmp_wait_release.h:978)
==167396==    by 0x63A762BE: __kmp_hyper_barrier_release(barrier_type, kmp_info*, int, int, int, void*) (kmp_barrier.cpp:1191)
==167396==    by 0x63A79E56: __kmp_fork_barrier(int, int) (kmp_barrier.cpp:2547)
==167396==    by 0x63AC233C: __kmp_launch_thread (kmp_runtime.cpp:6712)
==167396==    by 0x63B36420: __kmp_launch_worker(void*) (z_Linux_util.cpp:560)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x7D311000 0x7DB10FFF] client SP: 0x7DB0FAB0
valgrind stack range: [0x10174DA000 0x10175D9FFF] top usage: 4360 of 1048576

Thread 37: status = VgTs_WaitSys syscall 24 (lwpid 167618)
==167396==    at 0x4AA3C9B: sched_yield (syscall-template.S:120)
==167396==    by 0x63A7BEDD: __kmp_wait_template<kmp_flag_64<false, true>, true, false, true> (kmp_wait_release.h:610)
==167396==    by 0x63A7BEDD: kmp_flag_64<false, true>::wait(kmp_info*, int, void*) (kmp_wait_release.h:978)
==167396==    by 0x63A762BE: __kmp_hyper_barrier_release(barrier_type, kmp_info*, int, int, int, void*) (kmp_barrier.cpp:1191)
==167396==    by 0x63A79E56: __kmp_fork_barrier(int, int) (kmp_barrier.cpp:2547)
==167396==    by 0x63AC233C: __kmp_launch_thread (kmp_runtime.cpp:6712)
==167396==    by 0x63B36420: __kmp_launch_worker(void*) (z_Linux_util.cpp:560)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x7DB13000 0x7E312FFF] client SP: 0x7E311AE8
valgrind stack range: [0x10177DE000 0x10178DDFFF] top usage: 4360 of 1048576

Thread 38: status = VgTs_Yielding (lwpid 167620)
==167396==    at 0x63A7BE92: __kmp_wait_template<kmp_flag_64<false, true>, true, false, true> (kmp_wait_release.h:610)
==167396==    by 0x63A7BE92: kmp_flag_64<false, true>::wait(kmp_info*, int, void*) (kmp_wait_release.h:978)
==167396==    by 0x63A762BE: __kmp_hyper_barrier_release(barrier_type, kmp_info*, int, int, int, void*) (kmp_barrier.cpp:1191)
==167396==    by 0x63A79E56: __kmp_fork_barrier(int, int) (kmp_barrier.cpp:2547)
==167396==    by 0x63AC233C: __kmp_launch_thread (kmp_runtime.cpp:6712)
==167396==    by 0x63B36420: __kmp_launch_worker(void*) (z_Linux_util.cpp:560)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x7E315000 0x7EB14FFF] client SP: 0x7EB13B30
valgrind stack range: [0x1017AE6000 0x1017BE5FFF] top usage: 4360 of 1048576

Thread 39: status = VgTs_WaitSys syscall 24 (lwpid 167621)
==167396==    at 0x4AA3C9B: sched_yield (syscall-template.S:120)
==167396==    by 0x63A7BEDD: __kmp_wait_template<kmp_flag_64<false, true>, true, false, true> (kmp_wait_release.h:610)
==167396==    by 0x63A7BEDD: kmp_flag_64<false, true>::wait(kmp_info*, int, void*) (kmp_wait_release.h:978)
==167396==    by 0x63A762BE: __kmp_hyper_barrier_release(barrier_type, kmp_info*, int, int, int, void*) (kmp_barrier.cpp:1191)
==167396==    by 0x63A79E56: __kmp_fork_barrier(int, int) (kmp_barrier.cpp:2547)
==167396==    by 0x63AC233C: __kmp_launch_thread (kmp_runtime.cpp:6712)
==167396==    by 0x63B36420: __kmp_launch_worker(void*) (z_Linux_util.cpp:560)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x7EB17000 0x7F316FFF] client SP: 0x7F315B68
valgrind stack range: [0x1017DEA000 0x1017EE9FFF] top usage: 4360 of 1048576

Thread 40: status = VgTs_WaitSys syscall 24 (lwpid 167622)
==167396==    at 0x4AA3C9B: sched_yield (syscall-template.S:120)
==167396==    by 0x63A7BEDD: __kmp_wait_template<kmp_flag_64<false, true>, true, false, true> (kmp_wait_release.h:610)
==167396==    by 0x63A7BEDD: kmp_flag_64<false, true>::wait(kmp_info*, int, void*) (kmp_wait_release.h:978)
==167396==    by 0x63A762BE: __kmp_hyper_barrier_release(barrier_type, kmp_info*, int, int, int, void*) (kmp_barrier.cpp:1191)
==167396==    by 0x63A79E56: __kmp_fork_barrier(int, int) (kmp_barrier.cpp:2547)
==167396==    by 0x63AC233C: __kmp_launch_thread (kmp_runtime.cpp:6712)
==167396==    by 0x63B36420: __kmp_launch_worker(void*) (z_Linux_util.cpp:560)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x7F319000 0x7FB18FFF] client SP: 0x7FB17BA8
valgrind stack range: [0x10180EE000 0x10181EDFFF] top usage: 4360 of 1048576

Thread 41: status = VgTs_Yielding (lwpid 167623)
==167396==    at 0x63A7BE92: __kmp_wait_template<kmp_flag_64<false, true>, true, false, true> (kmp_wait_release.h:610)
==167396==    by 0x63A7BE92: kmp_flag_64<false, true>::wait(kmp_info*, int, void*) (kmp_wait_release.h:978)
==167396==    by 0x63A762BE: __kmp_hyper_barrier_release(barrier_type, kmp_info*, int, int, int, void*) (kmp_barrier.cpp:1191)
==167396==    by 0x63A79E56: __kmp_fork_barrier(int, int) (kmp_barrier.cpp:2547)
==167396==    by 0x63AC233C: __kmp_launch_thread (kmp_runtime.cpp:6712)
==167396==    by 0x63B36420: __kmp_launch_worker(void*) (z_Linux_util.cpp:560)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x7FB1B000 0x8031AFFF] client SP: 0x80319BF0
valgrind stack range: [0x10183F2000 0x10184F1FFF] top usage: 4360 of 1048576

Thread 42: status = VgTs_WaitSys syscall 24 (lwpid 167624)
==167396==    at 0x4AA3C9B: sched_yield (syscall-template.S:120)
==167396==    by 0x63A7BEDD: __kmp_wait_template<kmp_flag_64<false, true>, true, false, true> (kmp_wait_release.h:610)
==167396==    by 0x63A7BEDD: kmp_flag_64<false, true>::wait(kmp_info*, int, void*) (kmp_wait_release.h:978)
==167396==    by 0x63A762BE: __kmp_hyper_barrier_release(barrier_type, kmp_info*, int, int, int, void*) (kmp_barrier.cpp:1191)
==167396==    by 0x63A79E56: __kmp_fork_barrier(int, int) (kmp_barrier.cpp:2547)
==167396==    by 0x63AC233C: __kmp_launch_thread (kmp_runtime.cpp:6712)
==167396==    by 0x63B36420: __kmp_launch_worker(void*) (z_Linux_util.cpp:560)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x8031D000 0x80B1CFFF] client SP: 0x80B1BC28
valgrind stack range: [0x10186F6000 0x10187F5FFF] top usage: 4360 of 1048576

Thread 43: status = VgTs_WaitSys syscall 24 (lwpid 167625)
==167396==    at 0x4AA3C9B: sched_yield (syscall-template.S:120)
==167396==    by 0x63A7BEDD: __kmp_wait_template<kmp_flag_64<false, true>, true, false, true> (kmp_wait_release.h:610)
==167396==    by 0x63A7BEDD: kmp_flag_64<false, true>::wait(kmp_info*, int, void*) (kmp_wait_release.h:978)
==167396==    by 0x63A762BE: __kmp_hyper_barrier_release(barrier_type, kmp_info*, int, int, int, void*) (kmp_barrier.cpp:1191)
==167396==    by 0x63A79E56: __kmp_fork_barrier(int, int) (kmp_barrier.cpp:2547)
==167396==    by 0x63AC233C: __kmp_launch_thread (kmp_runtime.cpp:6712)
==167396==    by 0x63B36420: __kmp_launch_worker(void*) (z_Linux_util.cpp:560)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x80B1F000 0x8131EFFF] client SP: 0x8131DC68
valgrind stack range: [0x10189FA000 0x1018AF9FFF] top usage: 4872 of 1048576

Thread 44: status = VgTs_Yielding (lwpid 167626)
==167396==    at 0x63A7BE92: __kmp_wait_template<kmp_flag_64<false, true>, true, false, true> (kmp_wait_release.h:610)
==167396==    by 0x63A7BE92: kmp_flag_64<false, true>::wait(kmp_info*, int, void*) (kmp_wait_release.h:978)
==167396==    by 0x63A762BE: __kmp_hyper_barrier_release(barrier_type, kmp_info*, int, int, int, void*) (kmp_barrier.cpp:1191)
==167396==    by 0x63A79E56: __kmp_fork_barrier(int, int) (kmp_barrier.cpp:2547)
==167396==    by 0x63AC233C: __kmp_launch_thread (kmp_runtime.cpp:6712)
==167396==    by 0x63B36420: __kmp_launch_worker(void*) (z_Linux_util.cpp:560)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x81321000 0x81B20FFF] client SP: 0x81B1FCB0
valgrind stack range: [0x1018D02000 0x1018E01FFF] top usage: 4872 of 1048576

Thread 45: status = VgTs_Yielding (lwpid 167627)
==167396==    at 0x63A7BE92: __kmp_wait_template<kmp_flag_64<false, true>, true, false, true> (kmp_wait_release.h:610)
==167396==    by 0x63A7BE92: kmp_flag_64<false, true>::wait(kmp_info*, int, void*) (kmp_wait_release.h:978)
==167396==    by 0x63A762BE: __kmp_hyper_barrier_release(barrier_type, kmp_info*, int, int, int, void*) (kmp_barrier.cpp:1191)
==167396==    by 0x63A79E56: __kmp_fork_barrier(int, int) (kmp_barrier.cpp:2547)
==167396==    by 0x63AC233C: __kmp_launch_thread (kmp_runtime.cpp:6712)
==167396==    by 0x63B36420: __kmp_launch_worker(void*) (z_Linux_util.cpp:560)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x81B23000 0x82322FFF] client SP: 0x82321CF0
valgrind stack range: [0x1019006000 0x1019105FFF] top usage: 4872 of 1048576

Thread 46: status = VgTs_Yielding (lwpid 167628)
==167396==    at 0x63A7BE92: __kmp_wait_template<kmp_flag_64<false, true>, true, false, true> (kmp_wait_release.h:610)
==167396==    by 0x63A7BE92: kmp_flag_64<false, true>::wait(kmp_info*, int, void*) (kmp_wait_release.h:978)
==167396==    by 0x63A762BE: __kmp_hyper_barrier_release(barrier_type, kmp_info*, int, int, int, void*) (kmp_barrier.cpp:1191)
==167396==    by 0x63A79E56: __kmp_fork_barrier(int, int) (kmp_barrier.cpp:2547)
==167396==    by 0x63AC233C: __kmp_launch_thread (kmp_runtime.cpp:6712)
==167396==    by 0x63B36420: __kmp_launch_worker(void*) (z_Linux_util.cpp:560)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x82325000 0x82B24FFF] client SP: 0x82B23D30
valgrind stack range: [0x101930A000 0x1019409FFF] top usage: 4360 of 1048576

Thread 47: status = VgTs_Yielding (lwpid 167629)
==167396==    at 0x63A7BE92: __kmp_wait_template<kmp_flag_64<false, true>, true, false, true> (kmp_wait_release.h:610)
==167396==    by 0x63A7BE92: kmp_flag_64<false, true>::wait(kmp_info*, int, void*) (kmp_wait_release.h:978)
==167396==    by 0x63A762BE: __kmp_hyper_barrier_release(barrier_type, kmp_info*, int, int, int, void*) (kmp_barrier.cpp:1191)
==167396==    by 0x63A79E56: __kmp_fork_barrier(int, int) (kmp_barrier.cpp:2547)
==167396==    by 0x63AC233C: __kmp_launch_thread (kmp_runtime.cpp:6712)
==167396==    by 0x63B36420: __kmp_launch_worker(void*) (z_Linux_util.cpp:560)
==167396==    by 0x4A2FAC2: start_thread (pthread_create.c:442)
==167396==    by 0x4AC0A03: clone (clone.S:100)
client stack range: [0x82B27000 0x83326FFF] client SP: 0x83325D70
valgrind stack range: [0x1019612000 0x1019711FFF] top usage: 4872 of 1048576


Note: see also the FAQ in the source distribution.
It contains workarounds to several common problems.
In particular, if Valgrind aborted or crashed after
identifying problems in your program, there's a good chance
that fixing those problems will prevent Valgrind aborting or
crashing, especially if it happened in m_mallocfree.c.

If that doesn't help, please report this bug to: www.valgrind.org

In the bug report, send all the above text, the valgrind
version, and what OS and version you are using.  Thanks.

